<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.553">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Rohit Satyam">
<meta name="dcterms.date" content="2024-05-06">

<title>Reproducing JCI Paper</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="JCI-reanalysis_files/libs/clipboard/clipboard.min.js"></script>
<script src="JCI-reanalysis_files/libs/quarto-html/quarto.js"></script>
<script src="JCI-reanalysis_files/libs/quarto-html/popper.min.js"></script>
<script src="JCI-reanalysis_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="JCI-reanalysis_files/libs/quarto-html/anchor.min.js"></script>
<link href="JCI-reanalysis_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="JCI-reanalysis_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="JCI-reanalysis_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="JCI-reanalysis_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="JCI-reanalysis_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body>

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article toc-left">
<div id="quarto-sidebar-toc-left" class="sidebar toc-left">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Contents</h2>
   
  <ul>
  <li><a href="#bulk-rnaseq-data-analysis" id="toc-bulk-rnaseq-data-analysis" class="nav-link active" data-scroll-target="#bulk-rnaseq-data-analysis">Bulk RNASeq Data analysis</a>
  <ul class="collapse">
  <li><a href="#normalization-and-differential-expression-with-deseq" id="toc-normalization-and-differential-expression-with-deseq" class="nav-link" data-scroll-target="#normalization-and-differential-expression-with-deseq">Normalization and differential expression with DESeq</a></li>
  </ul></li>
  <li><a href="#understanding-the-de-results" id="toc-understanding-the-de-results" class="nav-link" data-scroll-target="#understanding-the-de-results">Understanding the DE results</a></li>
  <li><a href="#plot-ma" id="toc-plot-ma" class="nav-link" data-scroll-target="#plot-ma">plot MA</a>
  <ul class="collapse">
  <li><a href="#comparison-between-brauns-et.-al.-and-our-analysis." id="toc-comparison-between-brauns-et.-al.-and-our-analysis." class="nav-link" data-scroll-target="#comparison-between-brauns-et.-al.-and-our-analysis.">Comparison between <em>Brauns et. al.</em> and our analysis.</a></li>
  <li><a href="#gsea" id="toc-gsea" class="nav-link" data-scroll-target="#gsea">GSEA</a></li>
  <li><a href="#conclusion" id="toc-conclusion" class="nav-link" data-scroll-target="#conclusion">Conclusion</a></li>
  </ul></li>
  <li><a href="#session" id="toc-session" class="nav-link" data-scroll-target="#session">Session</a></li>
  </ul>
</nav>
</div>
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
</div>
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">Reproducing JCI Paper</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>
</div>


<div class="quarto-title-meta-author">
  <div class="quarto-title-meta-heading">Author</div>
  <div class="quarto-title-meta-heading">Affiliation</div>
  
    <div class="quarto-title-meta-contents">
    <p class="author">Rohit Satyam <a href="mailto:rohit.satyam@kaust.edu.sa" class="quarto-title-author-email"><i class="bi bi-envelope"></i></a> <a href="https://orcid.org/0000-0002-9311-4992" class="quarto-title-author-orcid"> <img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg=="></a></p>
  </div>
  <div class="quarto-title-meta-contents">
        <p class="affiliation">
            King Abdullah University of Science &amp; Technology, SA
          </p>
      </div>
  </div>

<div class="quarto-title-meta">

      
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">May 6, 2024</p>
    </div>
  </div>
  
    
  </div>
  


</header>


<section id="bulk-rnaseq-data-analysis" class="level2">
<h2 class="anchored" data-anchor-id="bulk-rnaseq-data-analysis">Bulk RNASeq Data analysis</h2>
<p>The raw bulk RNASeq data for 34 samples was obtained from <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE198256">GSE198256</a> and was uniformly processed using my in-house RNASeq analysis pipeline called <a href="https://github.com/Rohit-Satyam/RNAgrinder">RNAGrinder</a> (v1.0.1). We ran MultiQC on STAR output to get the sense of data quality post alignment. We found that the data was stranded and that reads came from Forward Strand. We will therefore take 2nd column of the STAR Gene count matrices.</p>
<div class="cell">
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DESeq2)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(NOISeq)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">'~/data/coursework2024/week1/jcpaper/results'</span>)</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>file_paths <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="at">path =</span> <span class="st">"."</span>,<span class="at">pattern =</span> <span class="st">"_ReadsPerGene.out.tab"</span>)</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Initialize an empty list to store the data</span></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>files<span class="ot">&lt;-</span><span class="fu">t</span>(plyr<span class="sc">::</span><span class="fu">ldply</span>(<span class="fu">lapply</span>(file_paths, <span class="cf">function</span>(x){</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">read.table</span>(x, <span class="at">skip=</span><span class="dv">4</span>)[,<span class="dv">3</span>]</span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>})))</span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(files) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"_ReadsPerGene.out.tab"</span>,<span class="st">""</span>,file_paths)</span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(files) <span class="ot">&lt;-</span> <span class="fu">read.table</span>(file_paths[<span class="dv">1</span>], <span class="at">skip=</span><span class="dv">4</span>)[,<span class="dv">1</span>]</span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a><span class="do">## reading metadata</span></span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a>meta <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"SraRunTable.txt"</span>, <span class="at">header =</span> T)</span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a>meta <span class="ot">&lt;-</span> meta[,<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">14</span>)]</span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a>meta<span class="sc">$</span>disease_state <span class="ot">&lt;-</span> mgsub<span class="sc">::</span><span class="fu">mgsub</span>(meta<span class="sc">$</span>disease_state,<span class="fu">c</span>(<span class="st">"Covid19: Acute infection"</span>,<span class="st">"Covid19: Recovery 3Mo"</span>,<span class="st">"Covid19: Recovery 6Mo"</span>), <span class="fu">c</span>(<span class="st">"Covid19AI"</span>,<span class="st">"Covid19_3Mo"</span>,<span class="st">"Covid19_6Mo"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>For making NOIseq ExpressionSet object, one can use the code given below to get the coordinates and GC% but since different database have different chromosome indexing (0-based or 1-based) this might not be ideal.</p>
<div class="cell">
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Using BioMart</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(biomaRt)</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>ensembl <span class="ot">&lt;-</span> <span class="fu">useEnsembl</span>(<span class="at">biomart =</span> <span class="st">"genes"</span>, <span class="at">dataset =</span> <span class="st">"hsapiens_gene_ensembl"</span>)</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a><span class="co">#View(listAttributes(ensembl))</span></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a><span class="do">## Get the coordinates of genes to get length and GC content </span></span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>attributes <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"ensembl_gene_id"</span>, <span class="st">"chromosome_name"</span>, <span class="st">"start_position"</span>,<span class="st">"end_position"</span>,<span class="st">"percentage_gene_gc_content"</span>)</span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>gene_info <span class="ot">&lt;-</span> <span class="fu">getBM</span>(<span class="at">attributes =</span> attributes, </span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>                   <span class="at">filters =</span> <span class="st">"ensembl_gene_id"</span>, </span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>                   <span class="at">values =</span> GeneStructureTools<span class="sc">::</span><span class="fu">removeVersion</span>(<span class="fu">rownames</span>(files)[<span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>]), </span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a>                   <span class="at">mart =</span> ensembl)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Let’s try to procure the same information using our own GTF and FASTA file that were used for mapping the reads. This will enable us to use this code for any other non-model organism for which such information can not be obtained from BioMart.</p>
<div class="cell">
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GenomicRanges)</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(rtracklayer)</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Rsamtools)</span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Load GTF file</span></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>GTFfile <span class="ot">&lt;-</span> <span class="st">"../../resource/gencode.v45.primary_assembly.annotation.gtf"</span></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>GTF <span class="ot">&lt;-</span> <span class="fu">import.gff</span>(GTFfile, <span class="at">format=</span><span class="st">"gtf"</span>, <span class="at">genome=</span><span class="st">"GRCh38"</span>, <span class="at">feature.type=</span><span class="st">"gene"</span>)</span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Reduce to merge overlapping features and simplify data structure</span></span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a>gene_ids <span class="ot">&lt;-</span> <span class="fu">mcols</span>(GTF)<span class="sc">$</span>gene_id</span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a>gene_biotypes <span class="ot">&lt;-</span> <span class="fu">mcols</span>(GTF)<span class="sc">$</span>gene_type</span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a><span class="do">## reading the FASTA reference</span></span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a>FASTAfile <span class="ot">=</span> <span class="st">"../../resource/GRCh38.primary_assembly.genome.fa"</span></span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a>FASTA <span class="ot">&lt;-</span> <span class="fu">FaFile</span>(FASTAfile)</span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a>seqs <span class="ot">&lt;-</span> <span class="fu">getSeq</span>(FASTA, GTF)  <span class="co"># Use original GTF ranges to ensure correct sequence retrieval</span></span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a>gc_counts <span class="ot">&lt;-</span> <span class="fu">letterFrequency</span>(seqs, <span class="at">letters =</span> <span class="fu">c</span>(<span class="st">"G"</span>, <span class="st">"C"</span>), <span class="at">as.prob =</span> <span class="cn">FALSE</span>)</span>
<span id="cb4-18"><a href="#cb4-18" aria-hidden="true" tabindex="-1"></a>lengths <span class="ot">&lt;-</span> <span class="fu">width</span>(GTF)</span>
<span id="cb4-19"><a href="#cb4-19" aria-hidden="true" tabindex="-1"></a>gc_content <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(gc_counts) <span class="sc">/</span> lengths</span>
<span id="cb4-20"><a href="#cb4-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-21"><a href="#cb4-21" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract necessary information (gene ID and gene biotype)</span></span>
<span id="cb4-22"><a href="#cb4-22" aria-hidden="true" tabindex="-1"></a>gene_info <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb4-23"><a href="#cb4-23" aria-hidden="true" tabindex="-1"></a>  <span class="at">gene_id =</span> gene_ids,</span>
<span id="cb4-24"><a href="#cb4-24" aria-hidden="true" tabindex="-1"></a>  <span class="at">gene_biotype =</span> gene_biotypes,</span>
<span id="cb4-25"><a href="#cb4-25" aria-hidden="true" tabindex="-1"></a>  <span class="at">gc=</span>gc_content,</span>
<span id="cb4-26"><a href="#cb4-26" aria-hidden="true" tabindex="-1"></a>  <span class="at">length=</span>lengths,<span class="at">starts=</span><span class="fu">start</span>(GTF),<span class="at">ends=</span><span class="fu">end</span>(GTF),</span>
<span id="cb4-27"><a href="#cb4-27" aria-hidden="true" tabindex="-1"></a>  <span class="at">chr=</span><span class="fu">seqnames</span>(GTF),<span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb4-28"><a href="#cb4-28" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(gene_info, <span class="st">"gene_info.rds"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Now, once we have all the attributes of all the genes, let’s compare the if the order of the genes in <code>gene_info</code> is same as the counts files we read.</p>
<div class="cell">
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(<span class="fu">rownames</span>(files)<span class="sc">==</span>gene_info<span class="sc">$</span>gene_id)</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(meta)<span class="ot">&lt;-</span>meta<span class="sc">$</span>Run</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>meta <span class="ot">&lt;-</span> meta[<span class="fu">colnames</span>(files),]</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(gene_info)<span class="ot">&lt;-</span>gene_info<span class="sc">$</span>gene_id</span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a><span class="do">## making expressionSet object to run noiseq functions</span></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>data_NOISEQ <span class="ot">&lt;-</span> <span class="fu">readData</span>(<span class="at">data =</span> files,</span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>                        <span class="at">length=</span><span class="fu">setNames</span>(gene_info<span class="sc">$</span>length, gene_info<span class="sc">$</span>gene_id),</span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a>                        <span class="at">gc=</span><span class="fu">setNames</span>(gene_info<span class="sc">$</span>gc, gene_info<span class="sc">$</span>gene_id),</span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a>                        <span class="at">biotype=</span><span class="fu">setNames</span>(gene_info<span class="sc">$</span>gene_biotype, gene_info<span class="sc">$</span>gene_id),</span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a>                        <span class="at">chromosome =</span> gene_info[,<span class="fu">c</span>(<span class="st">"chr"</span>,<span class="st">"starts"</span>,<span class="st">"ends"</span>)],</span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a>                        <span class="at">factors =</span> meta)</span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(data_NOISEQ,files,gene_info,<span class="at">file=</span><span class="st">"GSE198256_step1.Rda"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Loading all the </span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"GSE198256_step1.Rda"</span>)</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>myexplodata <span class="ot">&lt;-</span> <span class="fu">dat</span>(data_NOISEQ, <span class="at">type =</span> <span class="st">"biodetection"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Biotypes detection is to be computed for:
 [1] "SRR18280823" "SRR18280824" "SRR18280825" "SRR18280826" "SRR18280827"
 [6] "SRR18280828" "SRR18280829" "SRR18280830" "SRR18280831" "SRR18280832"
[11] "SRR18280833" "SRR18280834" "SRR18280835" "SRR18280836" "SRR18280837"
[16] "SRR18280838" "SRR18280839" "SRR18280840" "SRR18280841" "SRR18280842"
[21] "SRR18280843" "SRR18280844" "SRR18280845" "SRR18280846" "SRR18280847"
[26] "SRR18280848" "SRR18280849" "SRR18280850" "SRR18280851" "SRR18280852"
[31] "SRR18280853" "SRR18280854" "SRR18280855" "SRR18280856"</code></pre>
</div>
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="fu">explo.plot</span>(myexplodata, <span class="at">plottype =</span> <span class="st">"persample"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="JCI-reanalysis_files/figure-html/unnamed-chunk-6-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="JCI-reanalysis_files/figure-html/unnamed-chunk-6-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>))</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a><span class="fu">explo.plot</span>(myexplodata, <span class="at">samples =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>), <span class="at">toplot =</span> <span class="st">"protein_coding"</span>, <span class="at">plottype =</span> <span class="st">"comparison"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="JCI-reanalysis_files/figure-html/unnamed-chunk-6-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Percentage of protein_coding biotype in each sample:"
SRR18280823 SRR18280824 
    62.0294     61.2277 
[1] "Confidence interval at 95% for the difference of percentages: SRR18280823 - SRR18280824"
[1] 0.2641 1.3393
[1] "The percentage of this biotype is significantly DIFFERENT for these two samples (p-value = 0.003436 )."</code></pre>
</div>
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>mycountsbio <span class="ot">=</span> <span class="fu">dat</span>(data_NOISEQ, <span class="at">factor =</span> <span class="cn">NULL</span>, <span class="at">type =</span> <span class="st">"countsbio"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Warning: 24938 features with 0 counts in all samples are to be removed for this analysis."
[1] "Count distributions are to be computed for:"
 [1] "SRR18280823" "SRR18280824" "SRR18280825" "SRR18280826" "SRR18280827"
 [6] "SRR18280828" "SRR18280829" "SRR18280830" "SRR18280831" "SRR18280832"
[11] "SRR18280833" "SRR18280834" "SRR18280835" "SRR18280836" "SRR18280837"
[16] "SRR18280838" "SRR18280839" "SRR18280840" "SRR18280841" "SRR18280842"
[21] "SRR18280843" "SRR18280844" "SRR18280845" "SRR18280846" "SRR18280847"
[26] "SRR18280848" "SRR18280849" "SRR18280850" "SRR18280851" "SRR18280852"
[31] "SRR18280853" "SRR18280854" "SRR18280855" "SRR18280856"</code></pre>
</div>
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="fu">explo.plot</span>(mycountsbio, <span class="at">toplot =</span> <span class="dv">1</span>, <span class="at">samples =</span> <span class="dv">1</span>, <span class="at">plottype =</span> <span class="st">"boxplot"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="JCI-reanalysis_files/figure-html/unnamed-chunk-6-4.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>mysaturation <span class="ot">=</span> <span class="fu">dat</span>(data_NOISEQ, <span class="at">k =</span> <span class="dv">0</span>, <span class="at">ndepth =</span> <span class="dv">7</span>, <span class="at">type =</span> <span class="st">"saturation"</span>)</span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a><span class="fu">explo.plot</span>(mysaturation, <span class="at">toplot =</span> <span class="dv">1</span>, <span class="at">samples =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>, <span class="at">yleftlim =</span> <span class="cn">NULL</span>, <span class="at">yrightlim =</span> <span class="cn">NULL</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="JCI-reanalysis_files/figure-html/unnamed-chunk-6-5.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="fu">explo.plot</span>(mysaturation, <span class="at">toplot =</span> <span class="st">"protein_coding"</span>, <span class="at">samples =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="JCI-reanalysis_files/figure-html/unnamed-chunk-6-6.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="fu">explo.plot</span>(mycountsbio, <span class="at">toplot =</span> <span class="st">"protein_coding"</span>, <span class="at">samples =</span> <span class="cn">NULL</span>, <span class="at">plottype =</span> <span class="st">"boxplot"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="JCI-reanalysis_files/figure-html/unnamed-chunk-6-7.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="fu">explo.plot</span>(mycountsbio, <span class="at">toplot =</span> <span class="dv">1</span>, <span class="at">samples =</span> <span class="cn">NULL</span>, <span class="at">plottype =</span> <span class="st">"barplot"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="JCI-reanalysis_files/figure-html/unnamed-chunk-6-8.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>mylengthbias <span class="ot">=</span> <span class="fu">dat</span>(data_NOISEQ, <span class="at">factor =</span> <span class="st">"disease_state"</span>, <span class="at">type =</span> <span class="st">"lengthbias"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Warning: 24938 features with 0 counts in all samples are to be removed for this analysis."
[1] "Length bias detection information is to be computed for:"
[1] "Covid19_3Mo" "Covid19_6Mo" "Covid19AI"   "Healthy"    
[1] "Covid19_3Mo"

Call:
lm(formula = datos[, i] ~ bx)

Residuals:
    Min      1Q  Median      3Q     Max 
-9.4655 -0.8969 -0.1639  0.6966 15.5627 

Coefficients: (6 not defined because of singularities)
              Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)      7.848      3.111   2.523  0.01254 *  
bx1             -7.831      3.138  -2.495  0.01351 *  
bx2              4.219      3.330   1.267  0.20683    
bx3             14.568      3.579   4.070 7.12e-05 ***
bx4             16.982      3.685   4.608 7.82e-06 ***
bx5             24.332      3.995   6.091 7.01e-09 ***
bx6             27.949      4.451   6.279 2.63e-09 ***
bx7             35.840      5.288   6.778 1.81e-10 ***
bx8             20.344      7.503   2.712  0.00737 ** 
bx9             17.703     15.931   1.111  0.26799    
bx10            78.560     71.389   1.100  0.27265    
bx11          -595.484    739.037  -0.806  0.42148    
bx12          9754.187  11670.338   0.836  0.40441    
bx13        -82857.144  99360.674  -0.834  0.40548    
bx14        131334.042 157473.054   0.834  0.40542    
bx15         -2736.533   3332.830  -0.821  0.41272    
bx16                NA         NA      NA       NA    
bx17                NA         NA      NA       NA    
bx18                NA         NA      NA       NA    
bx19             8.715      4.831   1.804  0.07299 .  
bx20                NA         NA      NA       NA    
bx21                NA         NA      NA       NA    
bx22                NA         NA      NA       NA    
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 3.111 on 174 degrees of freedom
Multiple R-squared:  0.9372,    Adjusted R-squared:  0.9314 
F-statistic: 162.3 on 16 and 174 DF,  p-value: &lt; 2.2e-16

[1] "Covid19_6Mo"

Call:
lm(formula = datos[, i] ~ bx)

Residuals:
     Min       1Q   Median       3Q      Max 
-10.8769  -1.0367  -0.1446   0.4634  13.6736 

Coefficients: (6 not defined because of singularities)
              Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)  7.748e+00  3.169e+00   2.445  0.01548 *  
bx1         -7.828e+00  3.197e+00  -2.449  0.01533 *  
bx2          6.985e+00  3.392e+00   2.059  0.04098 *  
bx3          1.700e+01  3.646e+00   4.661 6.24e-06 ***
bx4          1.836e+01  3.754e+00   4.891 2.27e-06 ***
bx5          2.537e+01  4.070e+00   6.234 3.33e-09 ***
bx6          2.766e+01  4.534e+00   6.100 6.67e-09 ***
bx7          3.379e+01  5.386e+00   6.274 2.71e-09 ***
bx8          2.180e+01  7.643e+00   2.852  0.00487 ** 
bx9          9.942e+00  1.623e+01   0.613  0.54094    
bx10         1.034e+02  7.272e+01   1.422  0.15681    
bx11        -8.784e+02  7.529e+02  -1.167  0.24491    
bx12         1.421e+04  1.189e+04   1.195  0.23374    
bx13        -1.208e+05  1.012e+05  -1.193  0.23448    
bx14         1.914e+05  1.604e+05   1.193  0.23443    
bx15        -4.017e+03  3.395e+03  -1.183  0.23840    
bx16                NA         NA      NA       NA    
bx17                NA         NA      NA       NA    
bx18                NA         NA      NA       NA    
bx19         7.160e+00  4.922e+00   1.455  0.14752    
bx20                NA         NA      NA       NA    
bx21                NA         NA      NA       NA    
bx22                NA         NA      NA       NA    
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 3.169 on 174 degrees of freedom
Multiple R-squared:  0.9371,    Adjusted R-squared:  0.9313 
F-statistic:   162 on 16 and 174 DF,  p-value: &lt; 2.2e-16

[1] "Covid19AI"

Call:
lm(formula = datos[, i] ~ bx)

Residuals:
     Min       1Q   Median       3Q      Max 
-13.4852  -0.9321  -0.1135   0.8880  15.4956 

Coefficients: (6 not defined because of singularities)
              Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)  6.078e+00  3.304e+00   1.840  0.06750 .  
bx1         -6.165e+00  3.333e+00  -1.850  0.06604 .  
bx2          6.775e+00  3.537e+00   1.916  0.05704 .  
bx3          2.006e+01  3.801e+00   5.277 3.87e-07 ***
bx4          2.149e+01  3.914e+00   5.492 1.39e-07 ***
bx5          2.963e+01  4.243e+00   6.983 5.86e-11 ***
bx6          3.066e+01  4.727e+00   6.485 8.86e-10 ***
bx7          3.722e+01  5.615e+00   6.628 4.11e-10 ***
bx8          2.614e+01  7.968e+00   3.281  0.00125 ** 
bx9          1.474e+01  1.692e+01   0.871  0.38481    
bx10         1.014e+02  7.582e+01   1.338  0.18272    
bx11        -8.230e+02  7.849e+02  -1.049  0.29581    
bx12         1.335e+04  1.239e+04   1.077  0.28290    
bx13        -1.134e+05  1.055e+05  -1.075  0.28386    
bx14         1.798e+05  1.672e+05   1.075  0.28381    
bx15        -3.772e+03  3.540e+03  -1.066  0.28809    
bx16                NA         NA      NA       NA    
bx17                NA         NA      NA       NA    
bx18                NA         NA      NA       NA    
bx19         6.627e+00  5.131e+00   1.292  0.19820    
bx20                NA         NA      NA       NA    
bx21                NA         NA      NA       NA    
bx22                NA         NA      NA       NA    
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 3.304 on 174 degrees of freedom
Multiple R-squared:  0.939, Adjusted R-squared:  0.9334 
F-statistic: 167.4 on 16 and 174 DF,  p-value: &lt; 2.2e-16

[1] "Healthy"

Call:
lm(formula = datos[, i] ~ bx)

Residuals:
     Min       1Q   Median       3Q      Max 
-11.6418  -0.9323  -0.1420   0.6355  13.6420 

Coefficients: (6 not defined because of singularities)
              Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)  8.383e+00  3.065e+00   2.735  0.00688 ** 
bx1         -8.388e+00  3.092e+00  -2.713  0.00734 ** 
bx2          3.878e+00  3.281e+00   1.182  0.23877    
bx3          1.616e+01  3.526e+00   4.583 8.72e-06 ***
bx4          1.793e+01  3.630e+00   4.938 1.84e-06 ***
bx5          2.688e+01  3.936e+00   6.828 1.38e-10 ***
bx6          2.868e+01  4.385e+00   6.539 6.64e-10 ***
bx7          3.894e+01  5.209e+00   7.475 3.62e-12 ***
bx8          2.169e+01  7.392e+00   2.935  0.00379 ** 
bx9          1.212e+01  1.570e+01   0.772  0.44112    
bx10         1.103e+02  7.033e+01   1.569  0.11853    
bx11        -9.377e+02  7.281e+02  -1.288  0.19953    
bx12         1.517e+04  1.150e+04   1.319  0.18877    
bx13        -1.289e+05  9.789e+04  -1.317  0.18949    
bx14         2.044e+05  1.551e+05   1.317  0.18944    
bx15        -4.287e+03  3.284e+03  -1.305  0.19345    
bx16                NA         NA      NA       NA    
bx17                NA         NA      NA       NA    
bx18                NA         NA      NA       NA    
bx19         8.351e+00  4.760e+00   1.754  0.08112 .  
bx20                NA         NA      NA       NA    
bx21                NA         NA      NA       NA    
bx22                NA         NA      NA       NA    
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 3.065 on 174 degrees of freedom
Multiple R-squared:  0.9461,    Adjusted R-squared:  0.9411 
F-statistic: 190.8 on 16 and 174 DF,  p-value: &lt; 2.2e-16</code></pre>
</div>
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="fu">explo.plot</span>(mylengthbias, <span class="at">samples =</span> <span class="cn">NULL</span>, <span class="at">toplot =</span> <span class="st">"global"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="JCI-reanalysis_files/figure-html/unnamed-chunk-6-9.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>myGCbias <span class="ot">=</span> <span class="fu">dat</span>(data_NOISEQ, <span class="at">factor =</span> <span class="st">"disease_state"</span>, <span class="at">type =</span> <span class="st">"GCbias"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Warning: 24938 features with 0 counts in all samples are to be removed for this analysis."
[1] "GC content bias detection is to be computed for:"
[1] "Covid19_3Mo" "Covid19_6Mo" "Covid19AI"   "Healthy"    
[1] "Covid19_3Mo"

Call:
lm(formula = datos[, i] ~ bx)

Residuals:
   Min     1Q Median     3Q    Max 
-6.338 -1.925 -0.327  1.492  8.446 

Coefficients: (3 not defined because of singularities)
            Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)   0.8728     2.9122   0.300 0.764766    
bx1          -0.4246     4.1185  -0.103 0.917999    
bx2               NA         NA      NA       NA    
bx3               NA         NA      NA       NA    
bx4          45.8334   166.3560   0.276 0.783254    
bx5          -4.9602    17.1948  -0.288 0.773335    
bx6           0.4966     5.3013   0.094 0.925472    
bx7          23.2015     3.6153   6.418 1.31e-09 ***
bx8          15.8894     3.2878   4.833 2.98e-06 ***
bx9          11.7087     3.2223   3.634 0.000369 ***
bx10         12.2996     3.2228   3.816 0.000189 ***
bx11          5.2713     3.2583   1.618 0.107549    
bx12          8.1101     3.3074   2.452 0.015206 *  
bx13          4.5686     3.3462   1.365 0.173950    
bx14          9.5218     3.4065   2.795 0.005781 ** 
bx15          2.4138     3.5237   0.685 0.494257    
bx16          9.1736     3.7282   2.461 0.014864 *  
bx17          0.8047     4.0887   0.197 0.844201    
bx18          9.4458     4.8010   1.967 0.050745 .  
bx19         -0.3977     6.5520  -0.061 0.951672    
bx20          7.6692    33.1559   0.231 0.817352    
bx21        -46.8734   300.5608  -0.156 0.876254    
bx22              NA         NA      NA       NA    
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 2.912 on 171 degrees of freedom
Multiple R-squared:  0.7343,    Adjusted R-squared:  0.7048 
F-statistic: 24.87 on 19 and 171 DF,  p-value: &lt; 2.2e-16

[1] "Covid19_6Mo"

Call:
lm(formula = datos[, i] ~ bx)

Residuals:
    Min      1Q  Median      3Q     Max 
-6.7259 -1.6776 -0.2656  1.3023  7.1199 

Coefficients: (3 not defined because of singularities)
            Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)   1.2324     2.7550   0.447 0.655209    
bx1          -0.8126     3.8962  -0.209 0.835036    
bx2               NA         NA      NA       NA    
bx3               NA         NA      NA       NA    
bx4          82.7525   157.3771   0.526 0.599692    
bx5         -10.4934    16.2667  -0.645 0.519737    
bx6           2.6370     5.0152   0.526 0.599708    
bx7          23.1300     3.4201   6.763 2.05e-10 ***
bx8          15.4287     3.1103   4.960 1.69e-06 ***
bx9          12.7510     3.0484   4.183 4.59e-05 ***
bx10         11.8291     3.0489   3.880 0.000149 ***
bx11          6.2439     3.0824   2.026 0.044356 *  
bx12          8.1749     3.1289   2.613 0.009782 ** 
bx13          5.4184     3.1656   1.712 0.088776 .  
bx14         10.6355     3.2227   3.300 0.001176 ** 
bx15          3.1183     3.3335   0.935 0.350884    
bx16          9.9627     3.5270   2.825 0.005295 ** 
bx17          1.4943     3.8680   0.386 0.699740    
bx18         11.0157     4.5418   2.425 0.016333 *  
bx19         -0.4955     6.1983  -0.080 0.936373    
bx20         10.6953    31.3664   0.341 0.733538    
bx21        -77.0491   284.3383  -0.271 0.786736    
bx22              NA         NA      NA       NA    
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 2.755 on 171 degrees of freedom
Multiple R-squared:  0.738, Adjusted R-squared:  0.7089 
F-statistic: 25.36 on 19 and 171 DF,  p-value: &lt; 2.2e-16

[1] "Covid19AI"

Call:
lm(formula = datos[, i] ~ bx)

Residuals:
    Min      1Q  Median      3Q     Max 
-6.7296 -1.6842 -0.2066  1.3086  8.7798 

Coefficients: (3 not defined because of singularities)
            Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)   0.8060     2.9258   0.275 0.783277    
bx1          -0.3790     4.1377  -0.092 0.927125    
bx2               NA         NA      NA       NA    
bx3               NA         NA      NA       NA    
bx4          93.7803   167.1344   0.561 0.575459    
bx5         -12.2502    17.2752  -0.709 0.479217    
bx6           4.7012     5.3261   0.883 0.378651    
bx7          26.0874     3.6322   7.182 2.02e-11 ***
bx8          18.4137     3.3032   5.575 9.51e-08 ***
bx9          14.3476     3.2374   4.432 1.66e-05 ***
bx10         12.7815     3.2379   3.947 0.000115 ***
bx11          5.7420     3.2735   1.754 0.081213 .  
bx12          8.3205     3.3229   2.504 0.013216 *  
bx13          4.6085     3.3619   1.371 0.172230    
bx14          9.5206     3.4225   2.782 0.006013 ** 
bx15          2.9322     3.5402   0.828 0.408672    
bx16          7.4351     3.7457   1.985 0.048743 *  
bx17          1.5662     4.1078   0.381 0.703470    
bx18          8.0218     4.8234   1.663 0.098126 .  
bx19          0.4096     6.5826   0.062 0.950458    
bx20          4.3515    33.3111   0.131 0.896219    
bx21        -23.8193   301.9672  -0.079 0.937220    
bx22              NA         NA      NA       NA    
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 2.926 on 171 degrees of freedom
Multiple R-squared:  0.8071,    Adjusted R-squared:  0.7857 
F-statistic: 37.66 on 19 and 171 DF,  p-value: &lt; 2.2e-16

[1] "Healthy"

Call:
lm(formula = datos[, i] ~ bx)

Residuals:
    Min      1Q  Median      3Q     Max 
-7.6533 -1.7160 -0.1709  1.5009  7.4178 

Coefficients: (3 not defined because of singularities)
            Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)   0.7979     2.8117   0.284  0.77693    
bx1          -0.2920     3.9764  -0.073  0.94155    
bx2               NA         NA      NA       NA    
bx3               NA         NA      NA       NA    
bx4          90.6854   160.6176   0.565  0.57308    
bx5         -11.0290    16.6016  -0.664  0.50737    
bx6           3.7316     5.1185   0.729  0.46696    
bx7          26.2592     3.4906   7.523 2.91e-12 ***
bx8          17.6038     3.1744   5.546 1.09e-07 ***
bx9          13.4394     3.1112   4.320 2.64e-05 ***
bx10         12.5852     3.1116   4.045 7.92e-05 ***
bx11          6.0055     3.1459   1.909  0.05794 .  
bx12          8.1225     3.1933   2.544  0.01186 *  
bx13          4.9894     3.2308   1.544  0.12436    
bx14          9.1275     3.2890   2.775  0.00613 ** 
bx15          2.9866     3.4021   0.878  0.38125    
bx16          8.1541     3.5996   2.265  0.02475 *  
bx17          1.2585     3.9476   0.319  0.75027    
bx18          8.8529     4.6354   1.910  0.05783 .  
bx19         -0.3395     6.3260  -0.054  0.95726    
bx20          8.6621    32.0122   0.271  0.78703    
bx21        -61.6147   290.1931  -0.212  0.83211    
bx22              NA         NA      NA       NA    
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 2.812 on 171 degrees of freedom
Multiple R-squared:  0.805, Adjusted R-squared:  0.7833 
F-statistic: 37.14 on 19 and 171 DF,  p-value: &lt; 2.2e-16</code></pre>
</div>
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="fu">explo.plot</span>(myGCbias, <span class="at">samples =</span> <span class="cn">NULL</span>, <span class="at">toplot =</span> <span class="st">"global"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="JCI-reanalysis_files/figure-html/unnamed-chunk-6-10.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="co">#mycd = dat(data_NOISEQ, type = "cd", norm = FALSE, refColumn = 1)</span></span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a><span class="co">#explo.plot(mycd,samples = 1:12)</span></span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a>myPCA <span class="ot">=</span> <span class="fu">dat</span>(data_NOISEQ, <span class="at">type =</span> <span class="st">"PCA"</span>)</span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a><span class="do">## To match paper's label</span></span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a>myPCA<span class="sc">@</span>dat<span class="sc">$</span>factors<span class="sc">$</span>disease_state <span class="ot">&lt;-</span> mgsub<span class="sc">::</span><span class="fu">mgsub</span>(myPCA<span class="sc">@</span>dat<span class="sc">$</span>factors<span class="sc">$</span>disease_state,<span class="fu">unique</span>(myPCA<span class="sc">@</span>dat<span class="sc">$</span>factors<span class="sc">$</span>disease_state),<span class="fu">c</span>(<span class="st">"Late Recovery"</span>,<span class="st">"Early Recovery"</span>,<span class="st">"Acute"</span>,<span class="st">"Control"</span>))</span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a><span class="fu">explo.plot</span>(myPCA, <span class="at">factor =</span> <span class="st">"disease_state"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="JCI-reanalysis_files/figure-html/unnamed-chunk-6-11.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Since we have more than two condition, the linew below will throw an error</span></span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a><span class="co">#QCreport(data_NOISEQ, samples = NULL, factor = "disease_state", norm = FALSE)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="normalization-and-differential-expression-with-deseq" class="level3">
<h3 class="anchored" data-anchor-id="normalization-and-differential-expression-with-deseq">Normalization and differential expression with DESeq</h3>
<p>NOISeq offers different flavors of normalization such as RPKM, UQ and TMM. The code to do this is given below.</p>
<div class="cell">
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Normalisation approached provided by NOISeq includes RPKM, Upper Quartile and TMM (Trimmed Mean of M)</span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>myRPKM <span class="ot">=</span> <span class="fu">rpkm</span>(<span class="fu">assayData</span>(data_NOISEQ)<span class="sc">$</span>exprs, <span class="at">long =</span> data_NOISEQ<span class="sc">@</span>featureData<span class="sc">@</span>data<span class="sc">$</span>Length, <span class="at">k =</span> <span class="dv">0</span>, <span class="at">lc =</span> <span class="dv">1</span>)</span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a>myUQUA <span class="ot">=</span> <span class="fu">uqua</span>(<span class="fu">assayData</span>(data_NOISEQ)<span class="sc">$</span>exprs, <span class="at">long =</span> data_NOISEQ<span class="sc">@</span>featureData<span class="sc">@</span>data<span class="sc">$</span>Length, <span class="at">lc =</span> <span class="fl">0.5</span>, <span class="at">k =</span> <span class="dv">0</span>)</span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a>myTMM <span class="ot">=</span> <span class="fu">tmm</span>(<span class="fu">assayData</span>(data_NOISEQ)<span class="sc">$</span>exprs, <span class="at">long =</span> <span class="dv">1000</span>, <span class="at">lc =</span> <span class="dv">0</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>But since we wanna use DESeq2 we will use the raw counts and perform gene filtering before carrying out differential analysis.</p>
<div class="cell">
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Make DESeqDataSet object</span></span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a>GSE198256_DESeq2 <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> data_NOISEQ<span class="sc">@</span>assayData<span class="sc">$</span>exprs,</span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a>                                           <span class="at">colData =</span> data_NOISEQ<span class="sc">@</span>phenoData<span class="sc">@</span>data,</span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a>                                           <span class="at">design =</span> <span class="sc">~</span> disease_state)</span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a>GSE198256_DESeq2<span class="sc">$</span>disease_state<span class="ot">&lt;-</span><span class="fu">relevel</span>(GSE198256_DESeq2<span class="sc">$</span>disease_state,<span class="at">ref =</span> <span class="st">"Healthy"</span>)</span>
<span id="cb27-6"><a href="#cb27-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-7"><a href="#cb27-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Filtering genes with low counts</span></span>
<span id="cb27-8"><a href="#cb27-8" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(GSE198256_DESeq2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 63241    34</code></pre>
</div>
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Adding gene names and metadata</span></span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a><span class="fu">rowData</span>(GSE198256_DESeq2) <span class="ot">&lt;-</span> data_NOISEQ<span class="sc">@</span>featureData<span class="sc">@</span>data</span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a><span class="do">## Adding Gene symbols</span></span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a><span class="fu">mcols</span>(GSE198256_DESeq2)<span class="sc">$</span>names <span class="ot">&lt;-</span> GTF<span class="sc">$</span>gene_name[<span class="fu">match</span>(GTF<span class="sc">$</span>gene_id,<span class="fu">rownames</span>(data_NOISEQ<span class="sc">@</span>featureData<span class="sc">@</span>data) )]</span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-6"><a href="#cb29-6" aria-hidden="true" tabindex="-1"></a><span class="do">## Finding the smallest group size</span></span>
<span id="cb29-7"><a href="#cb29-7" aria-hidden="true" tabindex="-1"></a>smallestGroupSize <span class="ot">&lt;-</span> <span class="fu">min</span>(<span class="fu">table</span>(GSE198256_DESeq2<span class="sc">$</span>disease_state))</span>
<span id="cb29-8"><a href="#cb29-8" aria-hidden="true" tabindex="-1"></a>smallestGroupSize</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 6</code></pre>
</div>
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="co"># remove genes that do not have counts &gt;=10 in atleast 6 sample</span></span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a>keep <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(<span class="fu">counts</span>(GSE198256_DESeq2) <span class="sc">&gt;=</span> <span class="dv">10</span>) <span class="sc">&gt;=</span> smallestGroupSize</span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a>GSE198256_DESeq2_F <span class="ot">&lt;-</span> GSE198256_DESeq2[keep,] <span class="co"># 14,055 x 34</span></span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(GSE198256_DESeq2_F)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 15568    34</code></pre>
</div>
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a>GSE198256_DESeq2_F<span class="sc">$</span>labels <span class="ot">&lt;-</span> <span class="fu">as.character</span>(GSE198256_DESeq2_F<span class="sc">$</span>disease_state)</span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a>GSE198256_DESeq2_F<span class="sc">$</span>labels <span class="ot">&lt;-</span> mgsub<span class="sc">::</span><span class="fu">mgsub</span>(GSE198256_DESeq2_F<span class="sc">$</span>labels,<span class="fu">unique</span>(GSE198256_DESeq2_F<span class="sc">$</span>labels),<span class="fu">c</span>(<span class="st">"Late Recovery"</span>,<span class="st">"Early Recovery"</span>,<span class="st">"Acute"</span>,<span class="st">"Control"</span>))</span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a><span class="do">## making PCA using VST normalised data</span></span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a>vsd <span class="ot">&lt;-</span> <span class="fu">vst</span>(GSE198256_DESeq2_F, <span class="at">blind=</span><span class="cn">TRUE</span>,<span class="at">fitType =</span> <span class="st">"parametric"</span>)</span>
<span id="cb33-5"><a href="#cb33-5" aria-hidden="true" tabindex="-1"></a><span class="fu">plotPCA</span>(vsd, <span class="at">intgroup=</span><span class="fu">c</span>(<span class="st">"labels"</span>), <span class="at">returnData=</span><span class="cn">FALSE</span>)<span class="sc">+</span>ggplot2<span class="sc">::</span><span class="fu">theme_minimal</span>()<span class="sc">+</span></span>
<span id="cb33-6"><a href="#cb33-6" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">stat_ellipse</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="JCI-reanalysis_files/figure-html/unnamed-chunk-8-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Differential expression analysis</span></span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a>GSE198256_DESeq2_F<span class="ot">&lt;-</span> <span class="fu">DESeq</span>(GSE198256_DESeq2_F)</span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a><span class="co">#results(dds, contrast=c("condition","treated","untreated"))</span></span>
<span id="cb34-5"><a href="#cb34-5" aria-hidden="true" tabindex="-1"></a>ctrl.vs.acute <span class="ot">&lt;-</span> <span class="fu">results</span>(GSE198256_DESeq2_F, <span class="at">contrast=</span><span class="fu">c</span>(<span class="st">"disease_state"</span>,<span class="st">"Covid19AI"</span>,<span class="st">"Healthy"</span>))</span>
<span id="cb34-6"><a href="#cb34-6" aria-hidden="true" tabindex="-1"></a>ctrl.vs.early <span class="ot">&lt;-</span> <span class="fu">results</span>(GSE198256_DESeq2_F, <span class="at">contrast=</span><span class="fu">c</span>(<span class="st">"disease_state"</span>,<span class="st">"Covid19_3Mo"</span>,<span class="st">"Healthy"</span>))</span>
<span id="cb34-7"><a href="#cb34-7" aria-hidden="true" tabindex="-1"></a>ctrl.vs.late <span class="ot">&lt;-</span> <span class="fu">results</span>(GSE198256_DESeq2_F, <span class="at">contrast=</span><span class="fu">c</span>(<span class="st">"disease_state"</span>,<span class="st">"Covid19_6Mo"</span>,<span class="st">"Healthy"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="q1.-do-we-use-all-the-genes" class="level4">
<h4 class="anchored" data-anchor-id="q1.-do-we-use-all-the-genes">Q1. Do we use all the genes?</h4>
<p>Filtering genes before performing differential expression analysis is a critical step in RNASeq pipeline due to following reasons:</p>
<ul>
<li><p><strong>Low Expression</strong>: Genes that are expressed at very low levels across all samples often reflect background noise rather than true biological signals. These low counts can lead to high variability and unreliable statistical tests.</p></li>
<li><p><strong>Increasing Statistical Power</strong>: By removing genes that are unlikely to be informative (e.g., those with very low expression), you can reduce the multiple testing burden and increase the statistical power to detect true differences in the remaining genes.</p></li>
<li><p><strong>Speeding Up Analysis</strong>: Filtering out a significant number of genes from the analysis can dramatically decrease the computational time required for downstream statistical testing and analysis.</p></li>
<li><p><strong>Resource Management</strong>: This is particularly important when dealing with large datasets like TCGA Cancer dataset or limited computational resources.</p></li>
</ul>
</section>
<section id="q2.-how-do-we-select-which-ones" class="level4">
<h4 class="anchored" data-anchor-id="q2.-how-do-we-select-which-ones">Q2. How do we select which ones?</h4>
<p>There are several ways to skin a cat so you can choose one of the following criteria:</p>
<ul>
<li><strong>Expression Level</strong>: Genes with average counts below a certain threshold across all samples are often filtered out (See below that we take expression value threshold as <code>10</code>).</li>
<li><strong>Presence/Absence</strong>: Genes need to be expressed in a minimum number of samples or sample groups (See below where we choose the minimum no. of sample based on the smallest subgroup).</li>
<li><strong>Coefficient of Variation:</strong> Genes with very high variability relative to their mean expression might be excluded. This method is less common but one can filter genes which vary greatly across say technical or biological replicates.</li>
</ul>
</section>
</section>
</section>
<section id="understanding-the-de-results" class="level1">
<h1>Understanding the DE results</h1>
</section>
<section id="plot-ma" class="level1">
<h1>plot MA</h1>
<div class="cell">
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="fu">resultsNames</span>(GSE198256_DESeq2_F)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Intercept"                           
[2] "disease_state_Covid19_3Mo_vs_Healthy"
[3] "disease_state_Covid19_6Mo_vs_Healthy"
[4] "disease_state_Covid19AI_vs_Healthy"  </code></pre>
</div>
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a>ctrl.vs.acute <span class="ot">&lt;-</span> <span class="fu">lfcShrink</span>(GSE198256_DESeq2_F, <span class="at">coef =</span> <span class="dv">4</span>)</span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a>ctrl.vs.early <span class="ot">&lt;-</span> <span class="fu">lfcShrink</span>(GSE198256_DESeq2_F, <span class="at">coef=</span><span class="dv">2</span>)</span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a>ctrl.vs.late <span class="ot">&lt;-</span> <span class="fu">lfcShrink</span>(GSE198256_DESeq2_F, <span class="at">coef=</span><span class="dv">3</span>)</span>
<span id="cb37-4"><a href="#cb37-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-5"><a href="#cb37-5" aria-hidden="true" tabindex="-1"></a><span class="do">## table(rownames(ctrl.vs.acute)==rownames(rowData(GSE198256_DESeq2_F)))</span></span>
<span id="cb37-6"><a href="#cb37-6" aria-hidden="true" tabindex="-1"></a>ctrl.vs.acute<span class="sc">$</span>symbol <span class="ot">&lt;-</span> <span class="fu">rowData</span>(GSE198256_DESeq2_F)<span class="sc">$</span>names</span>
<span id="cb37-7"><a href="#cb37-7" aria-hidden="true" tabindex="-1"></a>ctrl.vs.early<span class="sc">$</span>symbol <span class="ot">&lt;-</span> <span class="fu">rowData</span>(GSE198256_DESeq2_F)<span class="sc">$</span>names</span>
<span id="cb37-8"><a href="#cb37-8" aria-hidden="true" tabindex="-1"></a>ctrl.vs.late<span class="sc">$</span>symbol <span class="ot">&lt;-</span> <span class="fu">rowData</span>(GSE198256_DESeq2_F)<span class="sc">$</span>names</span>
<span id="cb37-9"><a href="#cb37-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-10"><a href="#cb37-10" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb37-11"><a href="#cb37-11" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMA</span>(ctrl.vs.acute, <span class="at">ylim=</span><span class="fu">c</span>(<span class="sc">-</span><span class="dv">2</span>,<span class="dv">2</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="JCI-reanalysis_files/figure-html/unnamed-chunk-9-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMA</span>(ctrl.vs.early, <span class="at">ylim=</span><span class="fu">c</span>(<span class="sc">-</span><span class="dv">2</span>,<span class="dv">2</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="JCI-reanalysis_files/figure-html/unnamed-chunk-9-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMA</span>(ctrl.vs.late, <span class="at">ylim=</span><span class="fu">c</span>(<span class="sc">-</span><span class="dv">2</span>,<span class="dv">2</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="JCI-reanalysis_files/figure-html/unnamed-chunk-9-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Filter DEGs</span></span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a><span class="co">#filter DEGs: |FC| &gt; 2 and FDR &lt; 0.05 i.e. LFC value of 1</span></span>
<span id="cb40-3"><a href="#cb40-3" aria-hidden="true" tabindex="-1"></a>gtools<span class="sc">::</span><span class="fu">logratio2foldchange</span>(<span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 2</code></pre>
</div>
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a>ctrl.vs.acute.sig <span class="ot">&lt;-</span> <span class="fu">subset</span>(<span class="fu">as.data.frame</span>(ctrl.vs.acute) <span class="sc">%&gt;%</span>tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"Geneid"</span>), padj <span class="sc">&lt;=</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> <span class="fu">abs</span>(log2FoldChange)<span class="sc">&gt;</span><span class="dv">1</span>)</span>
<span id="cb42-2"><a href="#cb42-2" aria-hidden="true" tabindex="-1"></a>ctrl.vs.early.sig <span class="ot">&lt;-</span> <span class="fu">subset</span>(<span class="fu">as.data.frame</span>(ctrl.vs.early)<span class="sc">%&gt;%</span>tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"Geneid"</span>), padj <span class="sc">&lt;=</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> <span class="fu">abs</span>(log2FoldChange)<span class="sc">&gt;</span><span class="dv">1</span>)</span>
<span id="cb42-3"><a href="#cb42-3" aria-hidden="true" tabindex="-1"></a>ctrl.vs.late.sig <span class="ot">&lt;-</span> <span class="fu">subset</span>(<span class="fu">as.data.frame</span>(ctrl.vs.late)<span class="sc">%&gt;%</span>tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"Geneid"</span>), padj <span class="sc">&lt;=</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> <span class="fu">abs</span>(log2FoldChange)<span class="sc">&gt;</span><span class="dv">1</span>)</span>
<span id="cb42-4"><a href="#cb42-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-5"><a href="#cb42-5" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="st">"Ctrl vs Acute"</span><span class="ot">=</span>ctrl.vs.acute.sig,</span>
<span id="cb42-6"><a href="#cb42-6" aria-hidden="true" tabindex="-1"></a>            <span class="st">"Ctrl vs Early Recov"</span><span class="ot">=</span>ctrl.vs.early.sig,</span>
<span id="cb42-7"><a href="#cb42-7" aria-hidden="true" tabindex="-1"></a>            <span class="st">"Ctrl vs Late Recov"</span><span class="ot">=</span>ctrl.vs.late.sig)</span>
<span id="cb42-8"><a href="#cb42-8" aria-hidden="true" tabindex="-1"></a>writexl<span class="sc">::</span><span class="fu">write_xlsx</span>(res,<span class="st">"jci_DEGs.xlsx"</span>)</span>
<span id="cb42-9"><a href="#cb42-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Why to shrink: it looks at the largest fold changes that are not due </span></span>
<span id="cb42-10"><a href="#cb42-10" aria-hidden="true" tabindex="-1"></a><span class="co"># to low counts and uses these to inform a prior distribution. </span></span>
<span id="cb42-11"><a href="#cb42-11" aria-hidden="true" tabindex="-1"></a><span class="co"># So the large fold changes from genes with lots of statistical information are </span></span>
<span id="cb42-12"><a href="#cb42-12" aria-hidden="true" tabindex="-1"></a><span class="co"># not shrunk, while the imprecise fold changes are shrunk. This allows you to </span></span>
<span id="cb42-13"><a href="#cb42-13" aria-hidden="true" tabindex="-1"></a><span class="co"># compare all estimated LFC across experiments, for example, which is not really</span></span>
<span id="cb42-14"><a href="#cb42-14" aria-hidden="true" tabindex="-1"></a><span class="co"># feasible without the use of a prior. </span></span>
<span id="cb42-15"><a href="#cb42-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Michael Love https://support.bioconductor.org/p/77461/</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p><strong>Interpretation:</strong> The MA plot provides a quick visual assessment of the differential expression analysis. It helps to identify genes that are significantly differentially expressed (blue points) and to see how the fold changes distribute across genes with different expression levels. The data points’ spread along the y-axis (M) often narrows as you move to the right along the x-axis (A) because the relative change in expression (fold change) is generally more stable for genes with higher expression levels, reflecting the decreased variance associated with more abundant transcripts.</p>
<p>As expected, we can see that there are very few DEGs when Late Recovery monocytes samples were compared to control samples</p>
<section id="comparison-between-brauns-et.-al.-and-our-analysis." class="level3">
<h3 class="anchored" data-anchor-id="comparison-between-brauns-et.-al.-and-our-analysis.">Comparison between <em>Brauns et. al.</em> and our analysis.</h3>
<p>Severe COVID-19 significantly disrupts the myeloid immune cells during the infection, with effects lingering after recovery. The <span class="citation" data-cites="Brauns2022">(<a href="#ref-Brauns2022" role="doc-biblioref">Brauns et al. 2022</a>)</span> paper revealed that during the acute phase, patients with severe COVID-19 had monocytes with reduced cytokine production. Although the monocyte function mostly recovered after the illness, individuals who with Acute COVID showed heightened responses to certain immune triggers. Here we tried to reproduce their RNASeq data analysis in order to gauge to what extent we can reproduce the results. We use a slightly different pipeline and thresholds as tabulated below</p>
<p><strong>Difference in methodologies:</strong></p>
<table class="table">
<caption>(#tab:table) Differences between our RNASeq analysis pipeline and JCI paper’s.</caption>
<colgroup>
<col style="width: 25%">
<col style="width: 38%">
<col style="width: 36%">
</colgroup>
<thead>
<tr class="header">
<th>Steps</th>
<th>Bruns et al</th>
<th>Our Analysis</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Adapter Trimming</td>
<td>Trimmomatic-0.36 Truseq3-PE.fa:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36 HEADCROP:4</td>
<td>–detect_adapter_for_pe –qualified_quality_phred 30 –length_required 75 –correction</td>
</tr>
<tr class="even">
<td>Ribodepletion</td>
<td>NA</td>
<td>Used Ribodetector</td>
</tr>
<tr class="odd">
<td>Alignment</td>
<td>STAR 2.5.3a Default</td>
<td>STAR 2.7.11b using ENCODE guidelines</td>
</tr>
<tr class="even">
<td>Feature Counts</td>
<td>HTSeq-0.9.1 with “–nonunique all”</td>
<td>STAR GeneCount with HTSeq “–nonunique none” (default)</td>
</tr>
<tr class="odd">
<td>Gene Filtering</td>
<td>Gene counts &gt; 20; no. of samples &gt; 1</td>
<td>Gene counts &gt; 10; no. of samples &gt; 6</td>
</tr>
<tr class="even">
<td>Enrichment analysis</td>
<td>BubbleGUM</td>
<td>clusterProfiler</td>
</tr>
</tbody>
</table>
<p><strong>Comparing the results</strong></p>
<p>The PCA showed similar segregation of samples as reported by paper (see the plots below). We can see a clear separation between samples from acute COVID-19 patients and healthy controls. Samples from patients in the early recovery stage clustered distinctly, while those in the late recovery stage were embedded within the control group. PCA <strong>Plot A</strong> was plotted without using NOISeq, <strong>Plot B</strong> was plotted using filtered VST normalized counts and <strong>Plot C</strong> is from the paper.</p>
<div class="cell">
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(png)</span>
<span id="cb43-2"><a href="#cb43-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(grid)</span>
<span id="cb43-3"><a href="#cb43-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(gridExtra)</span>
<span id="cb43-4"><a href="#cb43-4" aria-hidden="true" tabindex="-1"></a>img3 <span class="ot">&lt;-</span>  <span class="fu">rasterGrob</span>(<span class="fu">as.raster</span>(<span class="fu">readPNG</span>(<span class="st">"images/Screenshot from 2024-04-27 18-07-36.png"</span>)), <span class="at">interpolate =</span> <span class="cn">FALSE</span>)</span>
<span id="cb43-5"><a href="#cb43-5" aria-hidden="true" tabindex="-1"></a>img1 <span class="ot">&lt;-</span>  <span class="fu">rasterGrob</span>(<span class="fu">as.raster</span>(<span class="fu">readPNG</span>(<span class="st">"images/PCA2.png"</span>)), <span class="at">interpolate =</span> <span class="cn">FALSE</span>)</span>
<span id="cb43-6"><a href="#cb43-6" aria-hidden="true" tabindex="-1"></a>img2 <span class="ot">&lt;-</span>  <span class="fu">rasterGrob</span>(<span class="fu">as.raster</span>(<span class="fu">readPNG</span>(<span class="st">"images/PCA3.png"</span>)), <span class="at">interpolate =</span> <span class="cn">FALSE</span>)</span>
<span id="cb43-7"><a href="#cb43-7" aria-hidden="true" tabindex="-1"></a>ggpubr<span class="sc">::</span><span class="fu">ggarrange</span>(img1,img2,img3, <span class="at">ncol =</span> <span class="dv">2</span>, <span class="at">nrow =</span> <span class="dv">2</span>,<span class="at">labels =</span> <span class="st">"AUTO"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="JCI-reanalysis_files/figure-html/unnamed-chunk-10-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>The authors used DESeq2 to perform the differential expression analysis. So using DESeq2 and performed DGEA for three comparisons: acute infection (AI) vs healthy (H), and the two stages of recovery (3Mo and 6Mo) vs healthy. We kept the genes that passed the thresholds mentioned in the article, FDR &lt; 0.05 and |FC| &gt; 2 i.e LFC of 1.</p>
<div class="cell">
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(schoolmath)</span>
<span id="cb44-2"><a href="#cb44-2" aria-hidden="true" tabindex="-1"></a>ctrl.vs.acute.sig<span class="sc">$</span>status <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.negative</span>(ctrl.vs.acute.sig<span class="sc">$</span>log2FoldChange),<span class="st">"down"</span>,<span class="st">"up"</span>)</span>
<span id="cb44-3"><a href="#cb44-3" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(ctrl.vs.acute.sig<span class="sc">$</span>status)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>
down   up 
 571  350 </code></pre>
</div>
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a>ctrl.vs.early.sig<span class="sc">$</span>status <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.negative</span>(ctrl.vs.early.sig<span class="sc">$</span>log2FoldChange),<span class="st">"down"</span>,<span class="st">"up"</span>)</span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(ctrl.vs.early.sig<span class="sc">$</span>status)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>
down   up 
 317  378 </code></pre>
</div>
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a>ctrl.vs.late.sig<span class="sc">$</span>status <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.negative</span>(ctrl.vs.late.sig<span class="sc">$</span>log2FoldChange),<span class="st">"down"</span>,<span class="st">"up"</span>)</span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(ctrl.vs.late.sig<span class="sc">$</span>status)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>
down   up 
  26   11 </code></pre>
</div>
</div>
<p>The patients with Acute COVID showed highest number of DEGs (<code>571 down-regulated and 350 up-regulated</code>) followed by Early Recovery patients (<code>317 down-regulated 378 up-regulated</code>). This was oppose to what paper observed i.e.&nbsp;more DEGs in Early recovery and less in Acute Infection patients. The Late recovery patients showed very few DEGs (<code>26 down 11 up</code>) which is in corcordance with the paper.</p>
<p>As authors reported, we also found down-regulation of genes related to key immune pathways, such as antigenic presentation, innate immune responses and MAPK and NF-κB signalling (JUNB, ATF3, NFkB2) in monocytes from patients suffering from Acute COVID-19. In addition, we also didn’t observe the upregulation of immunosuppressive genes such as ARG1, IL-10, or IDO1 in Acute Patients which is in concordance with what the paper has reported.</p>
<p>Like authors mentioned, for Ctrl vs Early recovery samples we also observed upregulation of genes encoding chemokines, along with important intracellular immunomodulatory proteins and transcription factors (PPARG, FOSL1, MAFB, MAFF, FOXO3) except ATF4.</p>
<div class="cell">
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb50"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(clusterProfiler)</span>
<span id="cb50-2"><a href="#cb50-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(org.Hs.eg.db)</span>
<span id="cb50-3"><a href="#cb50-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GeneStructureTools)</span>
<span id="cb50-4"><a href="#cb50-4" aria-hidden="true" tabindex="-1"></a>a <span class="ot">&lt;-</span> <span class="fu">enrichGO</span>(<span class="at">gene =</span> <span class="fu">unique</span>(<span class="fu">removeVersion</span>(ctrl.vs.acute.sig[ctrl.vs.acute.sig<span class="sc">$</span>status<span class="sc">==</span><span class="st">"up"</span>,]<span class="sc">$</span>Geneid)),</span>
<span id="cb50-5"><a href="#cb50-5" aria-hidden="true" tabindex="-1"></a><span class="at">OrgDb =</span> org.Hs.eg.db, <span class="at">keyType =</span> <span class="st">"ENSEMBL"</span>,</span>
<span id="cb50-6"><a href="#cb50-6" aria-hidden="true" tabindex="-1"></a><span class="at">ont =</span> <span class="st">"BP"</span>,</span>
<span id="cb50-7"><a href="#cb50-7" aria-hidden="true" tabindex="-1"></a><span class="at">pAdjustMethod =</span> <span class="st">"BH"</span>,</span>
<span id="cb50-8"><a href="#cb50-8" aria-hidden="true" tabindex="-1"></a><span class="at">pvalueCutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb50-9"><a href="#cb50-9" aria-hidden="true" tabindex="-1"></a><span class="at">qvalueCutoff =</span> <span class="fl">0.1</span>,</span>
<span id="cb50-10"><a href="#cb50-10" aria-hidden="true" tabindex="-1"></a><span class="at">readable =</span> <span class="cn">TRUE</span>)</span>
<span id="cb50-11"><a href="#cb50-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-12"><a href="#cb50-12" aria-hidden="true" tabindex="-1"></a>b <span class="ot">&lt;-</span> <span class="fu">enrichGO</span>(<span class="at">gene =</span> <span class="fu">unique</span>(<span class="fu">removeVersion</span>(ctrl.vs.acute.sig[ctrl.vs.acute.sig<span class="sc">$</span>status<span class="sc">==</span><span class="st">"down"</span>,]<span class="sc">$</span>Geneid)),</span>
<span id="cb50-13"><a href="#cb50-13" aria-hidden="true" tabindex="-1"></a><span class="at">OrgDb =</span> org.Hs.eg.db, <span class="at">keyType =</span> <span class="st">"ENSEMBL"</span>,</span>
<span id="cb50-14"><a href="#cb50-14" aria-hidden="true" tabindex="-1"></a><span class="at">ont =</span> <span class="st">"BP"</span>,</span>
<span id="cb50-15"><a href="#cb50-15" aria-hidden="true" tabindex="-1"></a><span class="at">pAdjustMethod =</span> <span class="st">"BH"</span>,</span>
<span id="cb50-16"><a href="#cb50-16" aria-hidden="true" tabindex="-1"></a><span class="at">pvalueCutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb50-17"><a href="#cb50-17" aria-hidden="true" tabindex="-1"></a><span class="at">qvalueCutoff =</span> <span class="fl">0.1</span>,</span>
<span id="cb50-18"><a href="#cb50-18" aria-hidden="true" tabindex="-1"></a><span class="at">readable =</span> <span class="cn">TRUE</span>)</span>
<span id="cb50-19"><a href="#cb50-19" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb50-20"><a href="#cb50-20" aria-hidden="true" tabindex="-1"></a><span class="co">#dotplot</span></span>
<span id="cb50-21"><a href="#cb50-21" aria-hidden="true" tabindex="-1"></a>plt1 <span class="ot">&lt;-</span> <span class="fu">dotplot</span>(a, <span class="at">showCategory =</span> <span class="dv">10</span>,</span>
<span id="cb50-22"><a href="#cb50-22" aria-hidden="true" tabindex="-1"></a><span class="at">font.size =</span> <span class="dv">10</span>,</span>
<span id="cb50-23"><a href="#cb50-23" aria-hidden="true" tabindex="-1"></a><span class="at">title =</span> <span class="st">"Acute.vs.Healthy (Upreg)"</span>)</span>
<span id="cb50-24"><a href="#cb50-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-25"><a href="#cb50-25" aria-hidden="true" tabindex="-1"></a>plt2 <span class="ot">&lt;-</span> <span class="fu">dotplot</span>(b, <span class="at">showCategory =</span> <span class="dv">10</span>,</span>
<span id="cb50-26"><a href="#cb50-26" aria-hidden="true" tabindex="-1"></a><span class="at">font.size =</span> <span class="dv">10</span>,</span>
<span id="cb50-27"><a href="#cb50-27" aria-hidden="true" tabindex="-1"></a><span class="at">title =</span> <span class="st">"Acute.vs.Healthy (Downreg)"</span>)</span>
<span id="cb50-28"><a href="#cb50-28" aria-hidden="true" tabindex="-1"></a>ggpubr<span class="sc">::</span><span class="fu">ggarrange</span>(plt1,plt2,<span class="at">labels =</span> <span class="st">"AUTO"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="JCI-reanalysis_files/figure-html/unnamed-chunk-12-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>ORA gave many over-represented biological terms for Acute DEGs but nothing significant for Eary Recovery Samples.</p>
<div class="cell">
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb51"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a>a <span class="ot">&lt;-</span> <span class="fu">enrichGO</span>(<span class="at">gene =</span> <span class="fu">unique</span>(<span class="fu">removeVersion</span>(ctrl.vs.early.sig[ctrl.vs.early.sig<span class="sc">$</span>status<span class="sc">==</span><span class="st">"up"</span>,]<span class="sc">$</span>Geneid)),</span>
<span id="cb51-2"><a href="#cb51-2" aria-hidden="true" tabindex="-1"></a><span class="at">OrgDb =</span> org.Hs.eg.db, <span class="at">keyType =</span> <span class="st">"ENSEMBL"</span>,</span>
<span id="cb51-3"><a href="#cb51-3" aria-hidden="true" tabindex="-1"></a><span class="at">ont =</span> <span class="st">"BP"</span>,</span>
<span id="cb51-4"><a href="#cb51-4" aria-hidden="true" tabindex="-1"></a><span class="at">pAdjustMethod =</span> <span class="st">"BH"</span>,</span>
<span id="cb51-5"><a href="#cb51-5" aria-hidden="true" tabindex="-1"></a><span class="at">pvalueCutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb51-6"><a href="#cb51-6" aria-hidden="true" tabindex="-1"></a><span class="at">qvalueCutoff =</span> <span class="fl">0.1</span>,</span>
<span id="cb51-7"><a href="#cb51-7" aria-hidden="true" tabindex="-1"></a><span class="at">readable =</span> <span class="cn">TRUE</span>)</span>
<span id="cb51-8"><a href="#cb51-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-9"><a href="#cb51-9" aria-hidden="true" tabindex="-1"></a>b <span class="ot">&lt;-</span> <span class="fu">enrichGO</span>(<span class="at">gene =</span> <span class="fu">unique</span>(<span class="fu">removeVersion</span>(ctrl.vs.early.sig[ctrl.vs.early.sig<span class="sc">$</span>status<span class="sc">==</span><span class="st">"down"</span>,]<span class="sc">$</span>Geneid)),</span>
<span id="cb51-10"><a href="#cb51-10" aria-hidden="true" tabindex="-1"></a><span class="at">OrgDb =</span> org.Hs.eg.db, <span class="at">keyType =</span> <span class="st">"ENSEMBL"</span>,</span>
<span id="cb51-11"><a href="#cb51-11" aria-hidden="true" tabindex="-1"></a><span class="at">ont =</span> <span class="st">"BP"</span>,</span>
<span id="cb51-12"><a href="#cb51-12" aria-hidden="true" tabindex="-1"></a><span class="at">pAdjustMethod =</span> <span class="st">"BH"</span>,</span>
<span id="cb51-13"><a href="#cb51-13" aria-hidden="true" tabindex="-1"></a><span class="at">pvalueCutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb51-14"><a href="#cb51-14" aria-hidden="true" tabindex="-1"></a><span class="at">qvalueCutoff =</span> <span class="fl">0.1</span>,</span>
<span id="cb51-15"><a href="#cb51-15" aria-hidden="true" tabindex="-1"></a><span class="at">readable =</span> <span class="cn">TRUE</span>)</span>
<span id="cb51-16"><a href="#cb51-16" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb51-17"><a href="#cb51-17" aria-hidden="true" tabindex="-1"></a><span class="co">#dotplot</span></span>
<span id="cb51-18"><a href="#cb51-18" aria-hidden="true" tabindex="-1"></a><span class="co"># plt1 &lt;- dotplot(a, showCategory = 10,</span></span>
<span id="cb51-19"><a href="#cb51-19" aria-hidden="true" tabindex="-1"></a><span class="co"># font.size = 10,</span></span>
<span id="cb51-20"><a href="#cb51-20" aria-hidden="true" tabindex="-1"></a><span class="co"># title = "Early Recov.vs.Healthy (Upreg)")</span></span>
<span id="cb51-21"><a href="#cb51-21" aria-hidden="true" tabindex="-1"></a><span class="do">## No-enrichment was observed for upregulated genes</span></span>
<span id="cb51-22"><a href="#cb51-22" aria-hidden="true" tabindex="-1"></a>plt2 <span class="ot">&lt;-</span> <span class="fu">dotplot</span>(b, <span class="at">showCategory =</span> <span class="dv">10</span>,</span>
<span id="cb51-23"><a href="#cb51-23" aria-hidden="true" tabindex="-1"></a><span class="at">font.size =</span> <span class="dv">10</span>,</span>
<span id="cb51-24"><a href="#cb51-24" aria-hidden="true" tabindex="-1"></a><span class="at">title =</span> <span class="st">"Early Recov.vs.Healthy (Downreg)"</span>)</span>
<span id="cb51-25"><a href="#cb51-25" aria-hidden="true" tabindex="-1"></a>ggpubr<span class="sc">::</span><span class="fu">ggarrange</span>(plt2,<span class="at">labels =</span> <span class="st">"AUTO"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="JCI-reanalysis_files/figure-html/unnamed-chunk-13-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="gsea" class="level3">
<h3 class="anchored" data-anchor-id="gsea">GSEA</h3>
<p>On performing GSEA analysis however, we got similar pathways to be enriched as shown by authors in the paper such as mitochondrial gene expression being enriched profoundly Acute vs.&nbsp;Healthy (AvH) (padj=8.684443e-06) relative to Early recovery cohort (EvH) (padj=0.013; not shown in figure). Instead of term “viral gene expression” we have related term “viral life cycle” in EvH (padj=5.931676e-05) but have smaller p-value in AvH (“response to virus” (padj=0.04712597)).</p>
<div class="cell">
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb52"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a>rank.wt <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(ctrl.vs.acute<span class="sc">$</span>pvalue) <span class="sc">*</span> <span class="fu">sign</span>(ctrl.vs.acute<span class="sc">$</span>log2FoldChange)</span>
<span id="cb52-2"><a href="#cb52-2" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(rank.wt) <span class="ot">&lt;-</span> <span class="fu">removeVersion</span>(<span class="fu">rownames</span>(ctrl.vs.acute))</span>
<span id="cb52-3"><a href="#cb52-3" aria-hidden="true" tabindex="-1"></a>rank.wt <span class="ot">&lt;-</span> <span class="fu">sort</span>(rank.wt,<span class="at">decreasing =</span> T)</span>
<span id="cb52-4"><a href="#cb52-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-5"><a href="#cb52-5" aria-hidden="true" tabindex="-1"></a>a <span class="ot">&lt;-</span> <span class="fu">gseGO</span>(<span class="at">geneList=</span>,rank.wt,</span>
<span id="cb52-6"><a href="#cb52-6" aria-hidden="true" tabindex="-1"></a><span class="at">ont =</span><span class="st">"BP"</span>,</span>
<span id="cb52-7"><a href="#cb52-7" aria-hidden="true" tabindex="-1"></a><span class="at">keyType =</span> <span class="st">"ENSEMBL"</span>,</span>
<span id="cb52-8"><a href="#cb52-8" aria-hidden="true" tabindex="-1"></a><span class="at">minGSSize =</span> <span class="dv">3</span>,</span>
<span id="cb52-9"><a href="#cb52-9" aria-hidden="true" tabindex="-1"></a><span class="at">maxGSSize =</span> <span class="dv">800</span>,</span>
<span id="cb52-10"><a href="#cb52-10" aria-hidden="true" tabindex="-1"></a><span class="at">pvalueCutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb52-11"><a href="#cb52-11" aria-hidden="true" tabindex="-1"></a><span class="at">verbose =</span> <span class="cn">TRUE</span>,</span>
<span id="cb52-12"><a href="#cb52-12" aria-hidden="true" tabindex="-1"></a><span class="at">OrgDb =</span> org.Hs.eg.db,</span>
<span id="cb52-13"><a href="#cb52-13" aria-hidden="true" tabindex="-1"></a><span class="at">pAdjustMethod =</span> <span class="st">"BH"</span>)</span>
<span id="cb52-14"><a href="#cb52-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-15"><a href="#cb52-15" aria-hidden="true" tabindex="-1"></a>plt1 <span class="ot">&lt;-</span> <span class="fu">dotplot</span>(a, <span class="at">showCategory =</span> <span class="dv">10</span>,</span>
<span id="cb52-16"><a href="#cb52-16" aria-hidden="true" tabindex="-1"></a>        <span class="at">font.size =</span> <span class="dv">10</span>,</span>
<span id="cb52-17"><a href="#cb52-17" aria-hidden="true" tabindex="-1"></a>        <span class="at">title =</span> <span class="st">"Acute.vs.Healthy"</span>)</span>
<span id="cb52-18"><a href="#cb52-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-19"><a href="#cb52-19" aria-hidden="true" tabindex="-1"></a>rank.wt <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(ctrl.vs.early<span class="sc">$</span>pvalue) <span class="sc">*</span> <span class="fu">sign</span>(ctrl.vs.early<span class="sc">$</span>log2FoldChange)</span>
<span id="cb52-20"><a href="#cb52-20" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(rank.wt) <span class="ot">&lt;-</span> <span class="fu">removeVersion</span>(<span class="fu">rownames</span>(ctrl.vs.early))</span>
<span id="cb52-21"><a href="#cb52-21" aria-hidden="true" tabindex="-1"></a>rank.wt <span class="ot">&lt;-</span> <span class="fu">sort</span>(rank.wt,<span class="at">decreasing =</span> T)</span>
<span id="cb52-22"><a href="#cb52-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-23"><a href="#cb52-23" aria-hidden="true" tabindex="-1"></a>b <span class="ot">&lt;-</span> <span class="fu">gseGO</span>(<span class="at">geneList=</span>,rank.wt,</span>
<span id="cb52-24"><a href="#cb52-24" aria-hidden="true" tabindex="-1"></a><span class="at">ont =</span><span class="st">"BP"</span>,</span>
<span id="cb52-25"><a href="#cb52-25" aria-hidden="true" tabindex="-1"></a><span class="at">keyType =</span> <span class="st">"ENSEMBL"</span>,</span>
<span id="cb52-26"><a href="#cb52-26" aria-hidden="true" tabindex="-1"></a><span class="at">minGSSize =</span> <span class="dv">3</span>,</span>
<span id="cb52-27"><a href="#cb52-27" aria-hidden="true" tabindex="-1"></a><span class="at">maxGSSize =</span> <span class="dv">800</span>,</span>
<span id="cb52-28"><a href="#cb52-28" aria-hidden="true" tabindex="-1"></a><span class="at">pvalueCutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb52-29"><a href="#cb52-29" aria-hidden="true" tabindex="-1"></a><span class="at">verbose =</span> <span class="cn">TRUE</span>,</span>
<span id="cb52-30"><a href="#cb52-30" aria-hidden="true" tabindex="-1"></a><span class="at">OrgDb =</span> org.Hs.eg.db,</span>
<span id="cb52-31"><a href="#cb52-31" aria-hidden="true" tabindex="-1"></a><span class="at">pAdjustMethod =</span> <span class="st">"BH"</span>)</span>
<span id="cb52-32"><a href="#cb52-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-33"><a href="#cb52-33" aria-hidden="true" tabindex="-1"></a>plt2 <span class="ot">&lt;-</span> <span class="fu">dotplot</span>(b, <span class="at">showCategory =</span> <span class="dv">10</span>,</span>
<span id="cb52-34"><a href="#cb52-34" aria-hidden="true" tabindex="-1"></a>        <span class="at">font.size =</span> <span class="dv">10</span>,</span>
<span id="cb52-35"><a href="#cb52-35" aria-hidden="true" tabindex="-1"></a>        <span class="at">title =</span> <span class="st">"Early Recov.vs.Healthy"</span>)</span>
<span id="cb52-36"><a href="#cb52-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-37"><a href="#cb52-37" aria-hidden="true" tabindex="-1"></a>ggpubr<span class="sc">::</span><span class="fu">ggarrange</span>(plt1,plt2,<span class="at">labels =</span> <span class="st">"AUTO"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="JCI-reanalysis_files/figure-html/unnamed-chunk-14-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Results from paper are shown below:</p>
<p><img src="jci.insight.154183.f7.jpg" class="img-fluid"></p>
</section>
<section id="conclusion" class="level2">
<h2 class="anchored" data-anchor-id="conclusion">Conclusion</h2>
<p>In summary, we were able to reproduce several major outcomes of the JCI paper but there was discord in number of DEGs when Acute and Early Recovery RNASeq data was compared to control. This could be due to number of reasons enlisted in @ref(tab:label). This should not be surprising because there are several ways to skin a cat and the outcome will differ from time to time based on which instruments were used.</p>
</section>
</section>
<section id="session" class="level1 unnumbered">
<h1 class="unnumbered">Session</h1>
<div class="cell">
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb53"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>R version 4.4.0 (2024-04-24)
Platform: x86_64-pc-linux-gnu
Running under: Ubuntu 20.04.6 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/atlas/libblas.so.3.10.3 
LAPACK: /usr/lib/x86_64-linux-gnu/atlas/liblapack.so.3.10.3;  LAPACK version 3.9.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

time zone: Asia/Riyadh
tzcode source: system (glibc)

attached base packages:
 [1] grid      splines   stats4    stats     graphics  grDevices utils    
 [8] datasets  methods   base     

other attached packages:
 [1] GeneStructureTools_1.25.0   org.Hs.eg.db_3.19.1        
 [3] AnnotationDbi_1.67.0        clusterProfiler_4.13.0     
 [5] schoolmath_0.4.2            gridExtra_2.3              
 [7] png_0.1-8                   Rsamtools_2.21.0           
 [9] Biostrings_2.73.0           XVector_0.45.0             
[11] rtracklayer_1.65.0          NOISeq_2.49.0              
[13] Matrix_1.7-0                DESeq2_1.45.0              
[15] SummarizedExperiment_1.35.0 Biobase_2.65.0             
[17] MatrixGenerics_1.17.0       matrixStats_1.3.0          
[19] GenomicRanges_1.57.0        GenomeInfoDb_1.41.0        
[21] IRanges_2.39.0              S4Vectors_0.43.0           
[23] BiocGenerics_0.51.0         dplyr_1.1.4                

loaded via a namespace (and not attached):
  [1] BiocIO_1.15.0                      filelock_1.0.3                    
  [3] bitops_1.0-7                       ggplotify_0.1.2                   
  [5] tibble_3.2.1                       polyclip_1.10-6                   
  [7] rpart_4.1.23                       XML_3.99-0.16.1                   
  [9] httr2_1.0.1                        lifecycle_1.0.4                   
 [11] rstatix_0.7.2                      ensembldb_2.29.0                  
 [13] lattice_0.22-6                     MASS_7.3-60.2                     
 [15] backports_1.4.1                    magrittr_2.0.3                    
 [17] Hmisc_5.1-2                        rmarkdown_2.26                    
 [19] yaml_2.3.8                         Gviz_1.49.0                       
 [21] cowplot_1.1.3                      DBI_1.2.2                         
 [23] RColorBrewer_1.1-3                 abind_1.4-5                       
 [25] zlibbioc_1.51.0                    purrr_1.0.2                       
 [27] AnnotationFilter_1.29.0            biovizBase_1.53.0                 
 [29] ggraph_2.2.1                       RCurl_1.98-1.14                   
 [31] nnet_7.3-19                        yulab.utils_0.1.4                 
 [33] VariantAnnotation_1.51.0           rappdirs_0.3.3                    
 [35] tweenr_2.0.3                       GenomeInfoDbData_1.2.12           
 [37] enrichplot_1.25.0                  ggrepel_0.9.5                     
 [39] tidytree_0.4.6                     codetools_0.2-20                  
 [41] DelayedArray_0.31.0                xml2_1.3.6                        
 [43] DOSE_3.31.0                        ggforce_0.4.2                     
 [45] tidyselect_1.2.1                   aplot_0.2.2                       
 [47] UCSC.utils_1.1.0                   farver_2.1.1                      
 [49] viridis_0.6.5                      base64enc_0.1-3                   
 [51] BiocFileCache_2.13.0               GenomicAlignments_1.41.0          
 [53] jsonlite_1.8.8                     Formula_1.2-5                     
 [55] tidygraph_1.3.1                    bbmle_1.0.25.1                    
 [57] progress_1.2.3                     tools_4.4.0                       
 [59] treeio_1.29.0                      stringdist_0.9.12                 
 [61] Rcpp_1.0.12                        glue_1.7.0                        
 [63] SparseArray_1.5.0                  xfun_0.43                         
 [65] qvalue_2.37.0                      withr_3.0.0                       
 [67] numDeriv_2016.8-1.1                BiocManager_1.30.23               
 [69] fastmap_1.1.1                      latticeExtra_0.6-30               
 [71] fansi_1.0.6                        digest_0.6.35                     
 [73] R6_2.5.1                           gridGraphics_0.5-1                
 [75] colorspace_2.1-0                   GO.db_3.19.1                      
 [77] gtools_3.9.5                       jpeg_0.1-10                       
 [79] dichromat_2.0-0.1                  biomaRt_2.61.0                    
 [81] RSQLite_2.3.6                      utf8_1.2.4                        
 [83] tidyr_1.3.1                        generics_0.1.3                    
 [85] data.table_1.15.4                  prettyunits_1.2.0                 
 [87] graphlayouts_1.1.1                 httr_1.4.7                        
 [89] htmlwidgets_1.6.4                  S4Arrays_1.5.0                    
 [91] scatterpie_0.2.2                   pkgconfig_2.0.3                   
 [93] gtable_0.3.5                       blob_1.2.4                        
 [95] mgsub_1.7.3                        shadowtext_0.1.3                  
 [97] htmltools_0.5.8.1                  carData_3.0-5                     
 [99] fgsea_1.31.0                       ProtGenerics_1.37.0               
[101] scales_1.3.0                       ggfun_0.1.4                       
[103] knitr_1.46                         rstudioapi_0.16.0                 
[105] reshape2_1.4.4                     rjson_0.2.21                      
[107] checkmate_2.3.1                    coda_0.19-4.1                     
[109] nlme_3.1-164                       curl_5.2.1                        
[111] bdsmatrix_1.3-7                    cachem_1.0.8                      
[113] stringr_1.5.1                      parallel_4.4.0                    
[115] HDO.db_0.99.1                      foreign_0.8-86                    
[117] restfulr_0.0.15                    apeglm_1.27.0                     
[119] pillar_1.9.0                       vctrs_0.6.5                       
[121] ggpubr_0.6.0                       car_3.1-2                         
[123] dbplyr_2.5.0                       cluster_2.1.6                     
[125] htmlTable_2.4.2                    evaluate_0.23                     
[127] GenomicFeatures_1.57.0             mvtnorm_1.2-4                     
[129] cli_3.6.2                          locfit_1.5-9.9                    
[131] compiler_4.4.0                     rlang_1.1.3                       
[133] crayon_1.5.2                       ggsignif_0.6.4                    
[135] labeling_0.4.3                     interp_1.1-6                      
[137] emdbook_1.3.13                     plyr_1.8.9                        
[139] fs_1.6.4                           writexl_1.5.0                     
[141] stringi_1.8.3                      deldir_2.0-4                      
[143] viridisLite_0.4.2                  BiocParallel_1.39.0               
[145] munsell_0.5.1                      lazyeval_0.2.2                    
[147] GOSemSim_2.31.0                    BSgenome_1.73.0                   
[149] hms_1.1.3                          patchwork_1.2.0                   
[151] bit64_4.0.5                        ggplot2_3.5.1                     
[153] KEGGREST_1.45.0                    igraph_2.0.3                      
[155] broom_1.0.5                        memoise_2.0.1                     
[157] ggtree_3.13.0                      fastmatch_1.1-4                   
[159] bit_4.0.5                          BSgenome.Mmusculus.UCSC.mm10_1.4.3
[161] ape_5.8                            gson_0.1.0                        </code></pre>
</div>
</div>
<!-- -->


</section>

<div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" role="doc-bibliography" id="quarto-bibliography"><h2 class="anchored quarto-appendix-heading">References</h2><div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list">
<div id="ref-Brauns2022" class="csl-entry" role="listitem">
Brauns, Elisa, Abdulkader Azouz, David Grimaldi, Hanxi Xiao, Séverine Thomas, Muriel Nguyen, Véronique Olislagers, et al. 2022. <span>“Functional Reprogramming of Monocytes in Patients with Acute and Convalescent Severe COVID-19.”</span> <em>JCI Insight</em> 7 (9). <a href="https://doi.org/10.1172/jci.insight.154183">https://doi.org/10.1172/jci.insight.154183</a>.
</div>
</div></section></div></main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb55" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb55-2"><a href="#cb55-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "Reproducing JCI Paper"</span></span>
<span id="cb55-3"><a href="#cb55-3" aria-hidden="true" tabindex="-1"></a><span class="an">author:</span></span>
<span id="cb55-4"><a href="#cb55-4" aria-hidden="true" tabindex="-1"></a><span class="co">  - name: Rohit Satyam</span></span>
<span id="cb55-5"><a href="#cb55-5" aria-hidden="true" tabindex="-1"></a><span class="co">    orcid: 0000-0002-9311-4992</span></span>
<span id="cb55-6"><a href="#cb55-6" aria-hidden="true" tabindex="-1"></a><span class="co">    email: rohit.satyam@kaust.edu.sa</span></span>
<span id="cb55-7"><a href="#cb55-7" aria-hidden="true" tabindex="-1"></a><span class="co">    affiliations:</span></span>
<span id="cb55-8"><a href="#cb55-8" aria-hidden="true" tabindex="-1"></a><span class="co">      - name: King Abdullah University of Science &amp; Technology, SA</span></span>
<span id="cb55-9"><a href="#cb55-9" aria-hidden="true" tabindex="-1"></a><span class="co">        address: 4700, KAUST</span></span>
<span id="cb55-10"><a href="#cb55-10" aria-hidden="true" tabindex="-1"></a><span class="co">        city: Thuwal</span></span>
<span id="cb55-11"><a href="#cb55-11" aria-hidden="true" tabindex="-1"></a><span class="co">        state: Jeedah</span></span>
<span id="cb55-12"><a href="#cb55-12" aria-hidden="true" tabindex="-1"></a><span class="co">        postal-code: 23955</span></span>
<span id="cb55-13"><a href="#cb55-13" aria-hidden="true" tabindex="-1"></a><span class="co">        </span></span>
<span id="cb55-14"><a href="#cb55-14" aria-hidden="true" tabindex="-1"></a><span class="an">clean:</span><span class="co"> false</span></span>
<span id="cb55-15"><a href="#cb55-15" aria-hidden="true" tabindex="-1"></a><span class="an">date:</span><span class="co"> "`r format(Sys.Date(), '%B %d, %Y')`"</span></span>
<span id="cb55-16"><a href="#cb55-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-17"><a href="#cb55-17" aria-hidden="true" tabindex="-1"></a><span class="an">vignette:</span><span class="co"> &gt;</span></span>
<span id="cb55-18"><a href="#cb55-18" aria-hidden="true" tabindex="-1"></a><span class="co"> %\VignetteIndexEntry{Reproducing JCI Paper}</span></span>
<span id="cb55-19"><a href="#cb55-19" aria-hidden="true" tabindex="-1"></a><span class="co"> %\VignetteEngine{knitr::rmarkdown}</span></span>
<span id="cb55-20"><a href="#cb55-20" aria-hidden="true" tabindex="-1"></a><span class="co"> %\VignetteEncoding{UTF-8}  </span></span>
<span id="cb55-21"><a href="#cb55-21" aria-hidden="true" tabindex="-1"></a><span class="co"> %\VignetteDepends{BiocStyle}</span></span>
<span id="cb55-22"><a href="#cb55-22" aria-hidden="true" tabindex="-1"></a><span class="co"> %\VignetteAuthor{Rohit Satyam}</span></span>
<span id="cb55-23"><a href="#cb55-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-24"><a href="#cb55-24" aria-hidden="true" tabindex="-1"></a><span class="an">format:</span><span class="co"> </span></span>
<span id="cb55-25"><a href="#cb55-25" aria-hidden="true" tabindex="-1"></a><span class="co">  html:</span></span>
<span id="cb55-26"><a href="#cb55-26" aria-hidden="true" tabindex="-1"></a><span class="co">    theme: yeti</span></span>
<span id="cb55-27"><a href="#cb55-27" aria-hidden="true" tabindex="-1"></a><span class="co">    toc: true</span></span>
<span id="cb55-28"><a href="#cb55-28" aria-hidden="true" tabindex="-1"></a><span class="co">    toc-title: Contents</span></span>
<span id="cb55-29"><a href="#cb55-29" aria-hidden="true" tabindex="-1"></a><span class="co">    toc-location: left</span></span>
<span id="cb55-30"><a href="#cb55-30" aria-hidden="true" tabindex="-1"></a><span class="co">    self-contained: false</span></span>
<span id="cb55-31"><a href="#cb55-31" aria-hidden="true" tabindex="-1"></a><span class="co">    code-fold: true</span></span>
<span id="cb55-32"><a href="#cb55-32" aria-hidden="true" tabindex="-1"></a><span class="co">    code-overflow: wrap</span></span>
<span id="cb55-33"><a href="#cb55-33" aria-hidden="true" tabindex="-1"></a><span class="co">    code-tools: true</span></span>
<span id="cb55-34"><a href="#cb55-34" aria-hidden="true" tabindex="-1"></a><span class="co">    code-summary: "Show the code"</span></span>
<span id="cb55-35"><a href="#cb55-35" aria-hidden="true" tabindex="-1"></a><span class="an">editor:</span><span class="co"> visual</span></span>
<span id="cb55-36"><a href="#cb55-36" aria-hidden="true" tabindex="-1"></a><span class="an">bibliography:</span><span class="co"> references.bib</span></span>
<span id="cb55-37"><a href="#cb55-37" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb55-38"><a href="#cb55-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-39"><a href="#cb55-39" aria-hidden="true" tabindex="-1"></a><span class="in">```{r setup, include=FALSE}</span></span>
<span id="cb55-40"><a href="#cb55-40" aria-hidden="true" tabindex="-1"></a><span class="in">knitr::opts_chunk$set(echo = TRUE, warning = F, message = F, cache = T,cache.lazy = FALSE)</span></span>
<span id="cb55-41"><a href="#cb55-41" aria-hidden="true" tabindex="-1"></a><span class="in">options(knitr.duplicate.label = "allow")</span></span>
<span id="cb55-42"><a href="#cb55-42" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb55-43"><a href="#cb55-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-44"><a href="#cb55-44" aria-hidden="true" tabindex="-1"></a><span class="fu">## Bulk RNASeq Data analysis</span></span>
<span id="cb55-45"><a href="#cb55-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-46"><a href="#cb55-46" aria-hidden="true" tabindex="-1"></a>The raw bulk RNASeq data for 34 samples was obtained from <span class="co">[</span><span class="ot">GSE198256</span><span class="co">](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE198256)</span> and was uniformly processed using my in-house RNASeq analysis pipeline called <span class="co">[</span><span class="ot">RNAGrinder</span><span class="co">](https://github.com/Rohit-Satyam/RNAgrinder)</span> (v1.0.1). We ran MultiQC on STAR output to get the sense of data quality post alignment. We found that the data was stranded and that reads came from Forward Strand. We will therefore take 2nd column of the STAR Gene count matrices.</span>
<span id="cb55-47"><a href="#cb55-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-50"><a href="#cb55-50" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb55-51"><a href="#cb55-51" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb55-52"><a href="#cb55-52" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DESeq2)</span>
<span id="cb55-53"><a href="#cb55-53" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(NOISeq)</span>
<span id="cb55-54"><a href="#cb55-54" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb55-55"><a href="#cb55-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-58"><a href="#cb55-58" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb55-59"><a href="#cb55-59" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">'~/data/coursework2024/week1/jcpaper/results'</span>)</span>
<span id="cb55-60"><a href="#cb55-60" aria-hidden="true" tabindex="-1"></a>file_paths <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="at">path =</span> <span class="st">"."</span>,<span class="at">pattern =</span> <span class="st">"_ReadsPerGene.out.tab"</span>)</span>
<span id="cb55-61"><a href="#cb55-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-62"><a href="#cb55-62" aria-hidden="true" tabindex="-1"></a><span class="co"># Initialize an empty list to store the data</span></span>
<span id="cb55-63"><a href="#cb55-63" aria-hidden="true" tabindex="-1"></a>files<span class="ot">&lt;-</span><span class="fu">t</span>(plyr<span class="sc">::</span><span class="fu">ldply</span>(<span class="fu">lapply</span>(file_paths, <span class="cf">function</span>(x){</span>
<span id="cb55-64"><a href="#cb55-64" aria-hidden="true" tabindex="-1"></a>  <span class="fu">read.table</span>(x, <span class="at">skip=</span><span class="dv">4</span>)[,<span class="dv">3</span>]</span>
<span id="cb55-65"><a href="#cb55-65" aria-hidden="true" tabindex="-1"></a>})))</span>
<span id="cb55-66"><a href="#cb55-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-67"><a href="#cb55-67" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(files) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"_ReadsPerGene.out.tab"</span>,<span class="st">""</span>,file_paths)</span>
<span id="cb55-68"><a href="#cb55-68" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(files) <span class="ot">&lt;-</span> <span class="fu">read.table</span>(file_paths[<span class="dv">1</span>], <span class="at">skip=</span><span class="dv">4</span>)[,<span class="dv">1</span>]</span>
<span id="cb55-69"><a href="#cb55-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-70"><a href="#cb55-70" aria-hidden="true" tabindex="-1"></a><span class="do">## reading metadata</span></span>
<span id="cb55-71"><a href="#cb55-71" aria-hidden="true" tabindex="-1"></a>meta <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"SraRunTable.txt"</span>, <span class="at">header =</span> T)</span>
<span id="cb55-72"><a href="#cb55-72" aria-hidden="true" tabindex="-1"></a>meta <span class="ot">&lt;-</span> meta[,<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">14</span>)]</span>
<span id="cb55-73"><a href="#cb55-73" aria-hidden="true" tabindex="-1"></a>meta<span class="sc">$</span>disease_state <span class="ot">&lt;-</span> mgsub<span class="sc">::</span><span class="fu">mgsub</span>(meta<span class="sc">$</span>disease_state,<span class="fu">c</span>(<span class="st">"Covid19: Acute infection"</span>,<span class="st">"Covid19: Recovery 3Mo"</span>,<span class="st">"Covid19: Recovery 6Mo"</span>), <span class="fu">c</span>(<span class="st">"Covid19AI"</span>,<span class="st">"Covid19_3Mo"</span>,<span class="st">"Covid19_6Mo"</span>))</span>
<span id="cb55-74"><a href="#cb55-74" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-75"><a href="#cb55-75" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb55-76"><a href="#cb55-76" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-77"><a href="#cb55-77" aria-hidden="true" tabindex="-1"></a>For making NOIseq ExpressionSet object, one can use the code given below to get the coordinates and GC% but since different database have different chromosome indexing (0-based or 1-based) this might not be ideal.</span>
<span id="cb55-78"><a href="#cb55-78" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-79"><a href="#cb55-79" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval=FALSE}</span></span>
<span id="cb55-80"><a href="#cb55-80" aria-hidden="true" tabindex="-1"></a><span class="in">## Using BioMart</span></span>
<span id="cb55-81"><a href="#cb55-81" aria-hidden="true" tabindex="-1"></a><span class="in">library(biomaRt)</span></span>
<span id="cb55-82"><a href="#cb55-82" aria-hidden="true" tabindex="-1"></a><span class="in">ensembl &lt;- useEnsembl(biomart = "genes", dataset = "hsapiens_gene_ensembl")</span></span>
<span id="cb55-83"><a href="#cb55-83" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-84"><a href="#cb55-84" aria-hidden="true" tabindex="-1"></a><span class="in">#View(listAttributes(ensembl))</span></span>
<span id="cb55-85"><a href="#cb55-85" aria-hidden="true" tabindex="-1"></a><span class="in">## Get the coordinates of genes to get length and GC content </span></span>
<span id="cb55-86"><a href="#cb55-86" aria-hidden="true" tabindex="-1"></a><span class="in">attributes &lt;- c("ensembl_gene_id", "chromosome_name", "start_position","end_position","percentage_gene_gc_content")</span></span>
<span id="cb55-87"><a href="#cb55-87" aria-hidden="true" tabindex="-1"></a><span class="in">gene_info &lt;- getBM(attributes = attributes, </span></span>
<span id="cb55-88"><a href="#cb55-88" aria-hidden="true" tabindex="-1"></a><span class="in">                   filters = "ensembl_gene_id", </span></span>
<span id="cb55-89"><a href="#cb55-89" aria-hidden="true" tabindex="-1"></a><span class="in">                   values = GeneStructureTools::removeVersion(rownames(files)[1:4]), </span></span>
<span id="cb55-90"><a href="#cb55-90" aria-hidden="true" tabindex="-1"></a><span class="in">                   mart = ensembl)</span></span>
<span id="cb55-91"><a href="#cb55-91" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb55-92"><a href="#cb55-92" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-93"><a href="#cb55-93" aria-hidden="true" tabindex="-1"></a>Let's try to procure the same information using our own GTF and FASTA file that were used for mapping the reads. This will enable us to use this code for any other non-model organism for which such information can not be obtained from BioMart.</span>
<span id="cb55-94"><a href="#cb55-94" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-95"><a href="#cb55-95" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval=TRUE}</span></span>
<span id="cb55-96"><a href="#cb55-96" aria-hidden="true" tabindex="-1"></a><span class="in">library(GenomicRanges)</span></span>
<span id="cb55-97"><a href="#cb55-97" aria-hidden="true" tabindex="-1"></a><span class="in">library(rtracklayer)</span></span>
<span id="cb55-98"><a href="#cb55-98" aria-hidden="true" tabindex="-1"></a><span class="in">library(Rsamtools)</span></span>
<span id="cb55-99"><a href="#cb55-99" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-100"><a href="#cb55-100" aria-hidden="true" tabindex="-1"></a><span class="in"># Load GTF file</span></span>
<span id="cb55-101"><a href="#cb55-101" aria-hidden="true" tabindex="-1"></a><span class="in">GTFfile &lt;- "../../resource/gencode.v45.primary_assembly.annotation.gtf"</span></span>
<span id="cb55-102"><a href="#cb55-102" aria-hidden="true" tabindex="-1"></a><span class="in">GTF &lt;- import.gff(GTFfile, format="gtf", genome="GRCh38", feature.type="gene")</span></span>
<span id="cb55-103"><a href="#cb55-103" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-104"><a href="#cb55-104" aria-hidden="true" tabindex="-1"></a><span class="in"># Reduce to merge overlapping features and simplify data structure</span></span>
<span id="cb55-105"><a href="#cb55-105" aria-hidden="true" tabindex="-1"></a><span class="in">gene_ids &lt;- mcols(GTF)$gene_id</span></span>
<span id="cb55-106"><a href="#cb55-106" aria-hidden="true" tabindex="-1"></a><span class="in">gene_biotypes &lt;- mcols(GTF)$gene_type</span></span>
<span id="cb55-107"><a href="#cb55-107" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-108"><a href="#cb55-108" aria-hidden="true" tabindex="-1"></a><span class="in">## reading the FASTA reference</span></span>
<span id="cb55-109"><a href="#cb55-109" aria-hidden="true" tabindex="-1"></a><span class="in">FASTAfile = "../../resource/GRCh38.primary_assembly.genome.fa"</span></span>
<span id="cb55-110"><a href="#cb55-110" aria-hidden="true" tabindex="-1"></a><span class="in">FASTA &lt;- FaFile(FASTAfile)</span></span>
<span id="cb55-111"><a href="#cb55-111" aria-hidden="true" tabindex="-1"></a><span class="in">seqs &lt;- getSeq(FASTA, GTF)  # Use original GTF ranges to ensure correct sequence retrieval</span></span>
<span id="cb55-112"><a href="#cb55-112" aria-hidden="true" tabindex="-1"></a><span class="in">gc_counts &lt;- letterFrequency(seqs, letters = c("G", "C"), as.prob = FALSE)</span></span>
<span id="cb55-113"><a href="#cb55-113" aria-hidden="true" tabindex="-1"></a><span class="in">lengths &lt;- width(GTF)</span></span>
<span id="cb55-114"><a href="#cb55-114" aria-hidden="true" tabindex="-1"></a><span class="in">gc_content &lt;- rowSums(gc_counts) / lengths</span></span>
<span id="cb55-115"><a href="#cb55-115" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-116"><a href="#cb55-116" aria-hidden="true" tabindex="-1"></a><span class="in"># Extract necessary information (gene ID and gene biotype)</span></span>
<span id="cb55-117"><a href="#cb55-117" aria-hidden="true" tabindex="-1"></a><span class="in">gene_info &lt;- data.frame(</span></span>
<span id="cb55-118"><a href="#cb55-118" aria-hidden="true" tabindex="-1"></a><span class="in">  gene_id = gene_ids,</span></span>
<span id="cb55-119"><a href="#cb55-119" aria-hidden="true" tabindex="-1"></a><span class="in">  gene_biotype = gene_biotypes,</span></span>
<span id="cb55-120"><a href="#cb55-120" aria-hidden="true" tabindex="-1"></a><span class="in">  gc=gc_content,</span></span>
<span id="cb55-121"><a href="#cb55-121" aria-hidden="true" tabindex="-1"></a><span class="in">  length=lengths,starts=start(GTF),ends=end(GTF),</span></span>
<span id="cb55-122"><a href="#cb55-122" aria-hidden="true" tabindex="-1"></a><span class="in">  chr=seqnames(GTF),stringsAsFactors = FALSE)</span></span>
<span id="cb55-123"><a href="#cb55-123" aria-hidden="true" tabindex="-1"></a><span class="in">saveRDS(gene_info, "gene_info.rds")</span></span>
<span id="cb55-124"><a href="#cb55-124" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb55-125"><a href="#cb55-125" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-126"><a href="#cb55-126" aria-hidden="true" tabindex="-1"></a>Now, once we have all the attributes of all the genes, let's compare the if the order of the genes in <span class="in">`gene_info`</span> is same as the counts files we read.</span>
<span id="cb55-127"><a href="#cb55-127" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-128"><a href="#cb55-128" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval=FALSE}</span></span>
<span id="cb55-129"><a href="#cb55-129" aria-hidden="true" tabindex="-1"></a><span class="in">table(rownames(files)==gene_info$gene_id)</span></span>
<span id="cb55-130"><a href="#cb55-130" aria-hidden="true" tabindex="-1"></a><span class="in">rownames(meta)&lt;-meta$Run</span></span>
<span id="cb55-131"><a href="#cb55-131" aria-hidden="true" tabindex="-1"></a><span class="in">meta &lt;- meta[colnames(files),]</span></span>
<span id="cb55-132"><a href="#cb55-132" aria-hidden="true" tabindex="-1"></a><span class="in">rownames(gene_info)&lt;-gene_info$gene_id</span></span>
<span id="cb55-133"><a href="#cb55-133" aria-hidden="true" tabindex="-1"></a><span class="in">## making expressionSet object to run noiseq functions</span></span>
<span id="cb55-134"><a href="#cb55-134" aria-hidden="true" tabindex="-1"></a><span class="in">data_NOISEQ &lt;- readData(data = files,</span></span>
<span id="cb55-135"><a href="#cb55-135" aria-hidden="true" tabindex="-1"></a><span class="in">                        length=setNames(gene_info$length, gene_info$gene_id),</span></span>
<span id="cb55-136"><a href="#cb55-136" aria-hidden="true" tabindex="-1"></a><span class="in">                        gc=setNames(gene_info$gc, gene_info$gene_id),</span></span>
<span id="cb55-137"><a href="#cb55-137" aria-hidden="true" tabindex="-1"></a><span class="in">                        biotype=setNames(gene_info$gene_biotype, gene_info$gene_id),</span></span>
<span id="cb55-138"><a href="#cb55-138" aria-hidden="true" tabindex="-1"></a><span class="in">                        chromosome = gene_info[,c("chr","starts","ends")],</span></span>
<span id="cb55-139"><a href="#cb55-139" aria-hidden="true" tabindex="-1"></a><span class="in">                        factors = meta)</span></span>
<span id="cb55-140"><a href="#cb55-140" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-141"><a href="#cb55-141" aria-hidden="true" tabindex="-1"></a><span class="in">save(data_NOISEQ,files,gene_info,file="GSE198256_step1.Rda")</span></span>
<span id="cb55-142"><a href="#cb55-142" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb55-143"><a href="#cb55-143" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-146"><a href="#cb55-146" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb55-147"><a href="#cb55-147" aria-hidden="true" tabindex="-1"></a><span class="do">## Loading all the </span></span>
<span id="cb55-148"><a href="#cb55-148" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"GSE198256_step1.Rda"</span>)</span>
<span id="cb55-149"><a href="#cb55-149" aria-hidden="true" tabindex="-1"></a>myexplodata <span class="ot">&lt;-</span> <span class="fu">dat</span>(data_NOISEQ, <span class="at">type =</span> <span class="st">"biodetection"</span>)</span>
<span id="cb55-150"><a href="#cb55-150" aria-hidden="true" tabindex="-1"></a><span class="fu">explo.plot</span>(myexplodata, <span class="at">plottype =</span> <span class="st">"persample"</span>)</span>
<span id="cb55-151"><a href="#cb55-151" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-152"><a href="#cb55-152" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>))</span>
<span id="cb55-153"><a href="#cb55-153" aria-hidden="true" tabindex="-1"></a><span class="fu">explo.plot</span>(myexplodata, <span class="at">samples =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>), <span class="at">toplot =</span> <span class="st">"protein_coding"</span>, <span class="at">plottype =</span> <span class="st">"comparison"</span>)</span>
<span id="cb55-154"><a href="#cb55-154" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-155"><a href="#cb55-155" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-156"><a href="#cb55-156" aria-hidden="true" tabindex="-1"></a>mycountsbio <span class="ot">=</span> <span class="fu">dat</span>(data_NOISEQ, <span class="at">factor =</span> <span class="cn">NULL</span>, <span class="at">type =</span> <span class="st">"countsbio"</span>)</span>
<span id="cb55-157"><a href="#cb55-157" aria-hidden="true" tabindex="-1"></a><span class="fu">explo.plot</span>(mycountsbio, <span class="at">toplot =</span> <span class="dv">1</span>, <span class="at">samples =</span> <span class="dv">1</span>, <span class="at">plottype =</span> <span class="st">"boxplot"</span>)</span>
<span id="cb55-158"><a href="#cb55-158" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-159"><a href="#cb55-159" aria-hidden="true" tabindex="-1"></a>mysaturation <span class="ot">=</span> <span class="fu">dat</span>(data_NOISEQ, <span class="at">k =</span> <span class="dv">0</span>, <span class="at">ndepth =</span> <span class="dv">7</span>, <span class="at">type =</span> <span class="st">"saturation"</span>)</span>
<span id="cb55-160"><a href="#cb55-160" aria-hidden="true" tabindex="-1"></a><span class="fu">explo.plot</span>(mysaturation, <span class="at">toplot =</span> <span class="dv">1</span>, <span class="at">samples =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>, <span class="at">yleftlim =</span> <span class="cn">NULL</span>, <span class="at">yrightlim =</span> <span class="cn">NULL</span>)</span>
<span id="cb55-161"><a href="#cb55-161" aria-hidden="true" tabindex="-1"></a><span class="fu">explo.plot</span>(mysaturation, <span class="at">toplot =</span> <span class="st">"protein_coding"</span>, <span class="at">samples =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>)</span>
<span id="cb55-162"><a href="#cb55-162" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-163"><a href="#cb55-163" aria-hidden="true" tabindex="-1"></a><span class="fu">explo.plot</span>(mycountsbio, <span class="at">toplot =</span> <span class="st">"protein_coding"</span>, <span class="at">samples =</span> <span class="cn">NULL</span>, <span class="at">plottype =</span> <span class="st">"boxplot"</span>)</span>
<span id="cb55-164"><a href="#cb55-164" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-165"><a href="#cb55-165" aria-hidden="true" tabindex="-1"></a><span class="fu">explo.plot</span>(mycountsbio, <span class="at">toplot =</span> <span class="dv">1</span>, <span class="at">samples =</span> <span class="cn">NULL</span>, <span class="at">plottype =</span> <span class="st">"barplot"</span>)</span>
<span id="cb55-166"><a href="#cb55-166" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-167"><a href="#cb55-167" aria-hidden="true" tabindex="-1"></a>mylengthbias <span class="ot">=</span> <span class="fu">dat</span>(data_NOISEQ, <span class="at">factor =</span> <span class="st">"disease_state"</span>, <span class="at">type =</span> <span class="st">"lengthbias"</span>)</span>
<span id="cb55-168"><a href="#cb55-168" aria-hidden="true" tabindex="-1"></a><span class="fu">explo.plot</span>(mylengthbias, <span class="at">samples =</span> <span class="cn">NULL</span>, <span class="at">toplot =</span> <span class="st">"global"</span>)</span>
<span id="cb55-169"><a href="#cb55-169" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-170"><a href="#cb55-170" aria-hidden="true" tabindex="-1"></a>myGCbias <span class="ot">=</span> <span class="fu">dat</span>(data_NOISEQ, <span class="at">factor =</span> <span class="st">"disease_state"</span>, <span class="at">type =</span> <span class="st">"GCbias"</span>)</span>
<span id="cb55-171"><a href="#cb55-171" aria-hidden="true" tabindex="-1"></a><span class="fu">explo.plot</span>(myGCbias, <span class="at">samples =</span> <span class="cn">NULL</span>, <span class="at">toplot =</span> <span class="st">"global"</span>)</span>
<span id="cb55-172"><a href="#cb55-172" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-173"><a href="#cb55-173" aria-hidden="true" tabindex="-1"></a><span class="co">#mycd = dat(data_NOISEQ, type = "cd", norm = FALSE, refColumn = 1)</span></span>
<span id="cb55-174"><a href="#cb55-174" aria-hidden="true" tabindex="-1"></a><span class="co">#explo.plot(mycd,samples = 1:12)</span></span>
<span id="cb55-175"><a href="#cb55-175" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-176"><a href="#cb55-176" aria-hidden="true" tabindex="-1"></a>myPCA <span class="ot">=</span> <span class="fu">dat</span>(data_NOISEQ, <span class="at">type =</span> <span class="st">"PCA"</span>)</span>
<span id="cb55-177"><a href="#cb55-177" aria-hidden="true" tabindex="-1"></a><span class="do">## To match paper's label</span></span>
<span id="cb55-178"><a href="#cb55-178" aria-hidden="true" tabindex="-1"></a>myPCA<span class="sc">@</span>dat<span class="sc">$</span>factors<span class="sc">$</span>disease_state <span class="ot">&lt;-</span> mgsub<span class="sc">::</span><span class="fu">mgsub</span>(myPCA<span class="sc">@</span>dat<span class="sc">$</span>factors<span class="sc">$</span>disease_state,<span class="fu">unique</span>(myPCA<span class="sc">@</span>dat<span class="sc">$</span>factors<span class="sc">$</span>disease_state),<span class="fu">c</span>(<span class="st">"Late Recovery"</span>,<span class="st">"Early Recovery"</span>,<span class="st">"Acute"</span>,<span class="st">"Control"</span>))</span>
<span id="cb55-179"><a href="#cb55-179" aria-hidden="true" tabindex="-1"></a><span class="fu">explo.plot</span>(myPCA, <span class="at">factor =</span> <span class="st">"disease_state"</span>)</span>
<span id="cb55-180"><a href="#cb55-180" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-181"><a href="#cb55-181" aria-hidden="true" tabindex="-1"></a><span class="do">## Since we have more than two condition, the linew below will throw an error</span></span>
<span id="cb55-182"><a href="#cb55-182" aria-hidden="true" tabindex="-1"></a><span class="co">#QCreport(data_NOISEQ, samples = NULL, factor = "disease_state", norm = FALSE)</span></span>
<span id="cb55-183"><a href="#cb55-183" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb55-184"><a href="#cb55-184" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-185"><a href="#cb55-185" aria-hidden="true" tabindex="-1"></a><span class="fu">### Normalization and differential expression with DESeq</span></span>
<span id="cb55-186"><a href="#cb55-186" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-187"><a href="#cb55-187" aria-hidden="true" tabindex="-1"></a>NOISeq offers different flavors of normalization such as RPKM, UQ and TMM. The code to do this is given below.</span>
<span id="cb55-188"><a href="#cb55-188" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-189"><a href="#cb55-189" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, message=FALSE, warning=FALSE, eval=F}</span></span>
<span id="cb55-190"><a href="#cb55-190" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-191"><a href="#cb55-191" aria-hidden="true" tabindex="-1"></a><span class="in">## Normalisation approached provided by NOISeq includes RPKM, Upper Quartile and TMM (Trimmed Mean of M)</span></span>
<span id="cb55-192"><a href="#cb55-192" aria-hidden="true" tabindex="-1"></a><span class="in">myRPKM = rpkm(assayData(data_NOISEQ)$exprs, long = data_NOISEQ@featureData@data$Length, k = 0, lc = 1)</span></span>
<span id="cb55-193"><a href="#cb55-193" aria-hidden="true" tabindex="-1"></a><span class="in">myUQUA = uqua(assayData(data_NOISEQ)$exprs, long = data_NOISEQ@featureData@data$Length, lc = 0.5, k = 0)</span></span>
<span id="cb55-194"><a href="#cb55-194" aria-hidden="true" tabindex="-1"></a><span class="in">myTMM = tmm(assayData(data_NOISEQ)$exprs, long = 1000, lc = 0)</span></span>
<span id="cb55-195"><a href="#cb55-195" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb55-196"><a href="#cb55-196" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-197"><a href="#cb55-197" aria-hidden="true" tabindex="-1"></a>But since we wanna use DESeq2 we will use the raw counts and perform gene filtering before carrying out differential analysis.</span>
<span id="cb55-198"><a href="#cb55-198" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-201"><a href="#cb55-201" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb55-202"><a href="#cb55-202" aria-hidden="true" tabindex="-1"></a><span class="do">## Make DESeqDataSet object</span></span>
<span id="cb55-203"><a href="#cb55-203" aria-hidden="true" tabindex="-1"></a>GSE198256_DESeq2 <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> data_NOISEQ<span class="sc">@</span>assayData<span class="sc">$</span>exprs,</span>
<span id="cb55-204"><a href="#cb55-204" aria-hidden="true" tabindex="-1"></a>                                           <span class="at">colData =</span> data_NOISEQ<span class="sc">@</span>phenoData<span class="sc">@</span>data,</span>
<span id="cb55-205"><a href="#cb55-205" aria-hidden="true" tabindex="-1"></a>                                           <span class="at">design =</span> <span class="sc">~</span> disease_state)</span>
<span id="cb55-206"><a href="#cb55-206" aria-hidden="true" tabindex="-1"></a>GSE198256_DESeq2<span class="sc">$</span>disease_state<span class="ot">&lt;-</span><span class="fu">relevel</span>(GSE198256_DESeq2<span class="sc">$</span>disease_state,<span class="at">ref =</span> <span class="st">"Healthy"</span>)</span>
<span id="cb55-207"><a href="#cb55-207" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-208"><a href="#cb55-208" aria-hidden="true" tabindex="-1"></a><span class="co"># Filtering genes with low counts</span></span>
<span id="cb55-209"><a href="#cb55-209" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(GSE198256_DESeq2)</span>
<span id="cb55-210"><a href="#cb55-210" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-211"><a href="#cb55-211" aria-hidden="true" tabindex="-1"></a><span class="do">## Adding gene names and metadata</span></span>
<span id="cb55-212"><a href="#cb55-212" aria-hidden="true" tabindex="-1"></a><span class="fu">rowData</span>(GSE198256_DESeq2) <span class="ot">&lt;-</span> data_NOISEQ<span class="sc">@</span>featureData<span class="sc">@</span>data</span>
<span id="cb55-213"><a href="#cb55-213" aria-hidden="true" tabindex="-1"></a><span class="do">## Adding Gene symbols</span></span>
<span id="cb55-214"><a href="#cb55-214" aria-hidden="true" tabindex="-1"></a><span class="fu">mcols</span>(GSE198256_DESeq2)<span class="sc">$</span>names <span class="ot">&lt;-</span> GTF<span class="sc">$</span>gene_name[<span class="fu">match</span>(GTF<span class="sc">$</span>gene_id,<span class="fu">rownames</span>(data_NOISEQ<span class="sc">@</span>featureData<span class="sc">@</span>data) )]</span>
<span id="cb55-215"><a href="#cb55-215" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-216"><a href="#cb55-216" aria-hidden="true" tabindex="-1"></a><span class="do">## Finding the smallest group size</span></span>
<span id="cb55-217"><a href="#cb55-217" aria-hidden="true" tabindex="-1"></a>smallestGroupSize <span class="ot">&lt;-</span> <span class="fu">min</span>(<span class="fu">table</span>(GSE198256_DESeq2<span class="sc">$</span>disease_state))</span>
<span id="cb55-218"><a href="#cb55-218" aria-hidden="true" tabindex="-1"></a>smallestGroupSize</span>
<span id="cb55-219"><a href="#cb55-219" aria-hidden="true" tabindex="-1"></a><span class="co"># remove genes that do not have counts &gt;=10 in atleast 6 sample</span></span>
<span id="cb55-220"><a href="#cb55-220" aria-hidden="true" tabindex="-1"></a>keep <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(<span class="fu">counts</span>(GSE198256_DESeq2) <span class="sc">&gt;=</span> <span class="dv">10</span>) <span class="sc">&gt;=</span> smallestGroupSize</span>
<span id="cb55-221"><a href="#cb55-221" aria-hidden="true" tabindex="-1"></a>GSE198256_DESeq2_F <span class="ot">&lt;-</span> GSE198256_DESeq2[keep,] <span class="co"># 14,055 x 34</span></span>
<span id="cb55-222"><a href="#cb55-222" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(GSE198256_DESeq2_F)</span>
<span id="cb55-223"><a href="#cb55-223" aria-hidden="true" tabindex="-1"></a>GSE198256_DESeq2_F<span class="sc">$</span>labels <span class="ot">&lt;-</span> <span class="fu">as.character</span>(GSE198256_DESeq2_F<span class="sc">$</span>disease_state)</span>
<span id="cb55-224"><a href="#cb55-224" aria-hidden="true" tabindex="-1"></a>GSE198256_DESeq2_F<span class="sc">$</span>labels <span class="ot">&lt;-</span> mgsub<span class="sc">::</span><span class="fu">mgsub</span>(GSE198256_DESeq2_F<span class="sc">$</span>labels,<span class="fu">unique</span>(GSE198256_DESeq2_F<span class="sc">$</span>labels),<span class="fu">c</span>(<span class="st">"Late Recovery"</span>,<span class="st">"Early Recovery"</span>,<span class="st">"Acute"</span>,<span class="st">"Control"</span>))</span>
<span id="cb55-225"><a href="#cb55-225" aria-hidden="true" tabindex="-1"></a><span class="do">## making PCA using VST normalised data</span></span>
<span id="cb55-226"><a href="#cb55-226" aria-hidden="true" tabindex="-1"></a>vsd <span class="ot">&lt;-</span> <span class="fu">vst</span>(GSE198256_DESeq2_F, <span class="at">blind=</span><span class="cn">TRUE</span>,<span class="at">fitType =</span> <span class="st">"parametric"</span>)</span>
<span id="cb55-227"><a href="#cb55-227" aria-hidden="true" tabindex="-1"></a><span class="fu">plotPCA</span>(vsd, <span class="at">intgroup=</span><span class="fu">c</span>(<span class="st">"labels"</span>), <span class="at">returnData=</span><span class="cn">FALSE</span>)<span class="sc">+</span>ggplot2<span class="sc">::</span><span class="fu">theme_minimal</span>()<span class="sc">+</span></span>
<span id="cb55-228"><a href="#cb55-228" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">stat_ellipse</span>()</span>
<span id="cb55-229"><a href="#cb55-229" aria-hidden="true" tabindex="-1"></a><span class="do">## Differential expression analysis</span></span>
<span id="cb55-230"><a href="#cb55-230" aria-hidden="true" tabindex="-1"></a>GSE198256_DESeq2_F<span class="ot">&lt;-</span> <span class="fu">DESeq</span>(GSE198256_DESeq2_F)</span>
<span id="cb55-231"><a href="#cb55-231" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-232"><a href="#cb55-232" aria-hidden="true" tabindex="-1"></a><span class="co">#results(dds, contrast=c("condition","treated","untreated"))</span></span>
<span id="cb55-233"><a href="#cb55-233" aria-hidden="true" tabindex="-1"></a>ctrl.vs.acute <span class="ot">&lt;-</span> <span class="fu">results</span>(GSE198256_DESeq2_F, <span class="at">contrast=</span><span class="fu">c</span>(<span class="st">"disease_state"</span>,<span class="st">"Covid19AI"</span>,<span class="st">"Healthy"</span>))</span>
<span id="cb55-234"><a href="#cb55-234" aria-hidden="true" tabindex="-1"></a>ctrl.vs.early <span class="ot">&lt;-</span> <span class="fu">results</span>(GSE198256_DESeq2_F, <span class="at">contrast=</span><span class="fu">c</span>(<span class="st">"disease_state"</span>,<span class="st">"Covid19_3Mo"</span>,<span class="st">"Healthy"</span>))</span>
<span id="cb55-235"><a href="#cb55-235" aria-hidden="true" tabindex="-1"></a>ctrl.vs.late <span class="ot">&lt;-</span> <span class="fu">results</span>(GSE198256_DESeq2_F, <span class="at">contrast=</span><span class="fu">c</span>(<span class="st">"disease_state"</span>,<span class="st">"Covid19_6Mo"</span>,<span class="st">"Healthy"</span>))</span>
<span id="cb55-236"><a href="#cb55-236" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb55-237"><a href="#cb55-237" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-238"><a href="#cb55-238" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Q1. Do we use all the genes?</span></span>
<span id="cb55-239"><a href="#cb55-239" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-240"><a href="#cb55-240" aria-hidden="true" tabindex="-1"></a>Filtering genes before performing differential expression analysis is a critical step in RNASeq pipeline due to following reasons:</span>
<span id="cb55-241"><a href="#cb55-241" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-242"><a href="#cb55-242" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**Low Expression**: Genes that are expressed at very low levels across all samples often reflect background noise rather than true biological signals. These low counts can lead to high variability and unreliable statistical tests.</span>
<span id="cb55-243"><a href="#cb55-243" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-244"><a href="#cb55-244" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**Increasing Statistical Power**: By removing genes that are unlikely to be informative (e.g., those with very low expression), you can reduce the multiple testing burden and increase the statistical power to detect true differences in the remaining genes.</span>
<span id="cb55-245"><a href="#cb55-245" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-246"><a href="#cb55-246" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**Speeding Up Analysis**: Filtering out a significant number of genes from the analysis can dramatically decrease the computational time required for downstream statistical testing and analysis.</span>
<span id="cb55-247"><a href="#cb55-247" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-248"><a href="#cb55-248" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**Resource Management**: This is particularly important when dealing with large datasets like TCGA Cancer dataset or limited computational resources.</span>
<span id="cb55-249"><a href="#cb55-249" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-250"><a href="#cb55-250" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Q2. How do we select which ones?</span></span>
<span id="cb55-251"><a href="#cb55-251" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-252"><a href="#cb55-252" aria-hidden="true" tabindex="-1"></a>There are several ways to skin a cat so you can choose one of the following criteria:</span>
<span id="cb55-253"><a href="#cb55-253" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-254"><a href="#cb55-254" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**Expression Level**: Genes with average counts below a certain threshold across all samples are often filtered out (See below that we take expression value threshold as <span class="in">`10`</span>).</span>
<span id="cb55-255"><a href="#cb55-255" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**Presence/Absence**: Genes need to be expressed in a minimum number of samples or sample groups (See below where we choose the minimum no. of sample based on the smallest subgroup).</span>
<span id="cb55-256"><a href="#cb55-256" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**Coefficient of Variation:** Genes with very high variability relative to their mean expression might be excluded. This method is less common but one can filter genes which vary greatly across say technical or biological replicates.</span>
<span id="cb55-257"><a href="#cb55-257" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-258"><a href="#cb55-258" aria-hidden="true" tabindex="-1"></a><span class="fu"># Understanding the DE results</span></span>
<span id="cb55-259"><a href="#cb55-259" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-260"><a href="#cb55-260" aria-hidden="true" tabindex="-1"></a><span class="fu"># plot MA</span></span>
<span id="cb55-261"><a href="#cb55-261" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-264"><a href="#cb55-264" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb55-265"><a href="#cb55-265" aria-hidden="true" tabindex="-1"></a><span class="fu">resultsNames</span>(GSE198256_DESeq2_F)</span>
<span id="cb55-266"><a href="#cb55-266" aria-hidden="true" tabindex="-1"></a>ctrl.vs.acute <span class="ot">&lt;-</span> <span class="fu">lfcShrink</span>(GSE198256_DESeq2_F, <span class="at">coef =</span> <span class="dv">4</span>)</span>
<span id="cb55-267"><a href="#cb55-267" aria-hidden="true" tabindex="-1"></a>ctrl.vs.early <span class="ot">&lt;-</span> <span class="fu">lfcShrink</span>(GSE198256_DESeq2_F, <span class="at">coef=</span><span class="dv">2</span>)</span>
<span id="cb55-268"><a href="#cb55-268" aria-hidden="true" tabindex="-1"></a>ctrl.vs.late <span class="ot">&lt;-</span> <span class="fu">lfcShrink</span>(GSE198256_DESeq2_F, <span class="at">coef=</span><span class="dv">3</span>)</span>
<span id="cb55-269"><a href="#cb55-269" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-270"><a href="#cb55-270" aria-hidden="true" tabindex="-1"></a><span class="do">## table(rownames(ctrl.vs.acute)==rownames(rowData(GSE198256_DESeq2_F)))</span></span>
<span id="cb55-271"><a href="#cb55-271" aria-hidden="true" tabindex="-1"></a>ctrl.vs.acute<span class="sc">$</span>symbol <span class="ot">&lt;-</span> <span class="fu">rowData</span>(GSE198256_DESeq2_F)<span class="sc">$</span>names</span>
<span id="cb55-272"><a href="#cb55-272" aria-hidden="true" tabindex="-1"></a>ctrl.vs.early<span class="sc">$</span>symbol <span class="ot">&lt;-</span> <span class="fu">rowData</span>(GSE198256_DESeq2_F)<span class="sc">$</span>names</span>
<span id="cb55-273"><a href="#cb55-273" aria-hidden="true" tabindex="-1"></a>ctrl.vs.late<span class="sc">$</span>symbol <span class="ot">&lt;-</span> <span class="fu">rowData</span>(GSE198256_DESeq2_F)<span class="sc">$</span>names</span>
<span id="cb55-274"><a href="#cb55-274" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-275"><a href="#cb55-275" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb55-276"><a href="#cb55-276" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMA</span>(ctrl.vs.acute, <span class="at">ylim=</span><span class="fu">c</span>(<span class="sc">-</span><span class="dv">2</span>,<span class="dv">2</span>))</span>
<span id="cb55-277"><a href="#cb55-277" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMA</span>(ctrl.vs.early, <span class="at">ylim=</span><span class="fu">c</span>(<span class="sc">-</span><span class="dv">2</span>,<span class="dv">2</span>))</span>
<span id="cb55-278"><a href="#cb55-278" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMA</span>(ctrl.vs.late, <span class="at">ylim=</span><span class="fu">c</span>(<span class="sc">-</span><span class="dv">2</span>,<span class="dv">2</span>))</span>
<span id="cb55-279"><a href="#cb55-279" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-280"><a href="#cb55-280" aria-hidden="true" tabindex="-1"></a><span class="do">## Filter DEGs</span></span>
<span id="cb55-281"><a href="#cb55-281" aria-hidden="true" tabindex="-1"></a><span class="co">#filter DEGs: |FC| &gt; 2 and FDR &lt; 0.05 i.e. LFC value of 1</span></span>
<span id="cb55-282"><a href="#cb55-282" aria-hidden="true" tabindex="-1"></a>gtools<span class="sc">::</span><span class="fu">logratio2foldchange</span>(<span class="dv">1</span>)</span>
<span id="cb55-283"><a href="#cb55-283" aria-hidden="true" tabindex="-1"></a>ctrl.vs.acute.sig <span class="ot">&lt;-</span> <span class="fu">subset</span>(<span class="fu">as.data.frame</span>(ctrl.vs.acute) <span class="sc">%&gt;%</span>tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"Geneid"</span>), padj <span class="sc">&lt;=</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> <span class="fu">abs</span>(log2FoldChange)<span class="sc">&gt;</span><span class="dv">1</span>)</span>
<span id="cb55-284"><a href="#cb55-284" aria-hidden="true" tabindex="-1"></a>ctrl.vs.early.sig <span class="ot">&lt;-</span> <span class="fu">subset</span>(<span class="fu">as.data.frame</span>(ctrl.vs.early)<span class="sc">%&gt;%</span>tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"Geneid"</span>), padj <span class="sc">&lt;=</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> <span class="fu">abs</span>(log2FoldChange)<span class="sc">&gt;</span><span class="dv">1</span>)</span>
<span id="cb55-285"><a href="#cb55-285" aria-hidden="true" tabindex="-1"></a>ctrl.vs.late.sig <span class="ot">&lt;-</span> <span class="fu">subset</span>(<span class="fu">as.data.frame</span>(ctrl.vs.late)<span class="sc">%&gt;%</span>tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"Geneid"</span>), padj <span class="sc">&lt;=</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> <span class="fu">abs</span>(log2FoldChange)<span class="sc">&gt;</span><span class="dv">1</span>)</span>
<span id="cb55-286"><a href="#cb55-286" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-287"><a href="#cb55-287" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="st">"Ctrl vs Acute"</span><span class="ot">=</span>ctrl.vs.acute.sig,</span>
<span id="cb55-288"><a href="#cb55-288" aria-hidden="true" tabindex="-1"></a>            <span class="st">"Ctrl vs Early Recov"</span><span class="ot">=</span>ctrl.vs.early.sig,</span>
<span id="cb55-289"><a href="#cb55-289" aria-hidden="true" tabindex="-1"></a>            <span class="st">"Ctrl vs Late Recov"</span><span class="ot">=</span>ctrl.vs.late.sig)</span>
<span id="cb55-290"><a href="#cb55-290" aria-hidden="true" tabindex="-1"></a>writexl<span class="sc">::</span><span class="fu">write_xlsx</span>(res,<span class="st">"jci_DEGs.xlsx"</span>)</span>
<span id="cb55-291"><a href="#cb55-291" aria-hidden="true" tabindex="-1"></a><span class="co"># Why to shrink: it looks at the largest fold changes that are not due </span></span>
<span id="cb55-292"><a href="#cb55-292" aria-hidden="true" tabindex="-1"></a><span class="co"># to low counts and uses these to inform a prior distribution. </span></span>
<span id="cb55-293"><a href="#cb55-293" aria-hidden="true" tabindex="-1"></a><span class="co"># So the large fold changes from genes with lots of statistical information are </span></span>
<span id="cb55-294"><a href="#cb55-294" aria-hidden="true" tabindex="-1"></a><span class="co"># not shrunk, while the imprecise fold changes are shrunk. This allows you to </span></span>
<span id="cb55-295"><a href="#cb55-295" aria-hidden="true" tabindex="-1"></a><span class="co"># compare all estimated LFC across experiments, for example, which is not really</span></span>
<span id="cb55-296"><a href="#cb55-296" aria-hidden="true" tabindex="-1"></a><span class="co"># feasible without the use of a prior. </span></span>
<span id="cb55-297"><a href="#cb55-297" aria-hidden="true" tabindex="-1"></a><span class="co"># Michael Love https://support.bioconductor.org/p/77461/</span></span>
<span id="cb55-298"><a href="#cb55-298" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb55-299"><a href="#cb55-299" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-300"><a href="#cb55-300" aria-hidden="true" tabindex="-1"></a>**Interpretation:** The MA plot provides a quick visual assessment of the differential expression analysis. It helps to identify genes that are significantly differentially expressed (blue points) and to see how the fold changes distribute across genes with different expression levels. The data points' spread along the y-axis (M) often narrows as you move to the right along the x-axis (A) because the relative change in expression (fold change) is generally more stable for genes with higher expression levels, reflecting the decreased variance associated with more abundant transcripts.</span>
<span id="cb55-301"><a href="#cb55-301" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-302"><a href="#cb55-302" aria-hidden="true" tabindex="-1"></a>As expected, we can see that there are very few DEGs when Late Recovery monocytes samples were compared to control samples</span>
<span id="cb55-303"><a href="#cb55-303" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-304"><a href="#cb55-304" aria-hidden="true" tabindex="-1"></a><span class="fu">### Comparison between *Brauns et. al.* and our analysis.</span></span>
<span id="cb55-305"><a href="#cb55-305" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-306"><a href="#cb55-306" aria-hidden="true" tabindex="-1"></a>Severe COVID-19 significantly disrupts the myeloid immune cells during the infection, with effects lingering after recovery. The <span class="co">[</span><span class="ot">@Brauns2022</span><span class="co">]</span> paper revealed that during the acute phase, patients with severe COVID-19 had monocytes with reduced cytokine production. Although the monocyte function mostly recovered after the illness, individuals who with Acute COVID showed heightened responses to certain immune triggers. Here we tried to reproduce their RNASeq data analysis in order to gauge to what extent we can reproduce the results. We use a slightly different pipeline and thresholds as tabulated below</span>
<span id="cb55-307"><a href="#cb55-307" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-308"><a href="#cb55-308" aria-hidden="true" tabindex="-1"></a>**Difference in methodologies:**</span>
<span id="cb55-309"><a href="#cb55-309" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-310"><a href="#cb55-310" aria-hidden="true" tabindex="-1"></a>| Steps               | Bruns et al                                                                                         | Our Analysis                                                                           |</span>
<span id="cb55-311"><a href="#cb55-311" aria-hidden="true" tabindex="-1"></a>|------------------|----------------------------|--------------------------|</span>
<span id="cb55-312"><a href="#cb55-312" aria-hidden="true" tabindex="-1"></a>| Adapter Trimming    | Trimmomatic-0.36 Truseq3-PE.fa:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36 HEADCROP:4 | --detect_adapter_for_pe --qualified_quality_phred 30 --length_required 75 --correction |</span>
<span id="cb55-313"><a href="#cb55-313" aria-hidden="true" tabindex="-1"></a>| Ribodepletion       | NA                                                                                                  | Used Ribodetector                                                                      |</span>
<span id="cb55-314"><a href="#cb55-314" aria-hidden="true" tabindex="-1"></a>| Alignment           | STAR 2.5.3a Default                                                                                 | STAR 2.7.11b using ENCODE guidelines                                                   |</span>
<span id="cb55-315"><a href="#cb55-315" aria-hidden="true" tabindex="-1"></a>| Feature Counts      | HTSeq-0.9.1 with “--nonunique all”                                                                  | STAR GeneCount with HTSeq “--nonunique none” (default)                                 |</span>
<span id="cb55-316"><a href="#cb55-316" aria-hidden="true" tabindex="-1"></a>| Gene Filtering      | Gene counts <span class="sc">\&gt;</span> 20; no. of samples <span class="sc">\&gt;</span> 1                                                              | Gene counts <span class="sc">\&gt;</span> 10; no. of samples <span class="sc">\&gt;</span> 6                                                 |</span>
<span id="cb55-317"><a href="#cb55-317" aria-hidden="true" tabindex="-1"></a>| Enrichment analysis | BubbleGUM                                                                                           | clusterProfiler                                                                        |</span>
<span id="cb55-318"><a href="#cb55-318" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-319"><a href="#cb55-319" aria-hidden="true" tabindex="-1"></a>: (<span class="sc">\#</span>tab:table) Differences between our RNASeq analysis pipeline and JCI paper's.</span>
<span id="cb55-320"><a href="#cb55-320" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-321"><a href="#cb55-321" aria-hidden="true" tabindex="-1"></a>**Comparing the results**</span>
<span id="cb55-322"><a href="#cb55-322" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-323"><a href="#cb55-323" aria-hidden="true" tabindex="-1"></a>The PCA showed similar segregation of samples as reported by paper (see the plots below). We can see a clear separation between samples from acute COVID-19 patients and healthy controls. Samples from patients in the early recovery stage clustered distinctly, while those in the late recovery stage were embedded within the control group. PCA **Plot A** was plotted without using NOISeq, **Plot B** was plotted using filtered VST normalized counts and **Plot C** is from the paper. </span>
<span id="cb55-324"><a href="#cb55-324" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-325"><a href="#cb55-325" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-328"><a href="#cb55-328" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb55-329"><a href="#cb55-329" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(png)</span>
<span id="cb55-330"><a href="#cb55-330" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(grid)</span>
<span id="cb55-331"><a href="#cb55-331" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(gridExtra)</span>
<span id="cb55-332"><a href="#cb55-332" aria-hidden="true" tabindex="-1"></a>img3 <span class="ot">&lt;-</span>  <span class="fu">rasterGrob</span>(<span class="fu">as.raster</span>(<span class="fu">readPNG</span>(<span class="st">"images/Screenshot from 2024-04-27 18-07-36.png"</span>)), <span class="at">interpolate =</span> <span class="cn">FALSE</span>)</span>
<span id="cb55-333"><a href="#cb55-333" aria-hidden="true" tabindex="-1"></a>img1 <span class="ot">&lt;-</span>  <span class="fu">rasterGrob</span>(<span class="fu">as.raster</span>(<span class="fu">readPNG</span>(<span class="st">"images/PCA2.png"</span>)), <span class="at">interpolate =</span> <span class="cn">FALSE</span>)</span>
<span id="cb55-334"><a href="#cb55-334" aria-hidden="true" tabindex="-1"></a>img2 <span class="ot">&lt;-</span>  <span class="fu">rasterGrob</span>(<span class="fu">as.raster</span>(<span class="fu">readPNG</span>(<span class="st">"images/PCA3.png"</span>)), <span class="at">interpolate =</span> <span class="cn">FALSE</span>)</span>
<span id="cb55-335"><a href="#cb55-335" aria-hidden="true" tabindex="-1"></a>ggpubr<span class="sc">::</span><span class="fu">ggarrange</span>(img1,img2,img3, <span class="at">ncol =</span> <span class="dv">2</span>, <span class="at">nrow =</span> <span class="dv">2</span>,<span class="at">labels =</span> <span class="st">"AUTO"</span>)</span>
<span id="cb55-336"><a href="#cb55-336" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb55-337"><a href="#cb55-337" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-338"><a href="#cb55-338" aria-hidden="true" tabindex="-1"></a>The authors used DESeq2 to perform the differential expression analysis. So using DESeq2 and performed DGEA for three comparisons: acute infection (AI) vs healthy (H), and the two stages of recovery (3Mo and 6Mo) vs healthy. We kept the genes that passed the thresholds mentioned in the article, FDR &lt; 0.05 and |FC| &gt; 2 i.e LFC of 1.</span>
<span id="cb55-339"><a href="#cb55-339" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-342"><a href="#cb55-342" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb55-343"><a href="#cb55-343" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(schoolmath)</span>
<span id="cb55-344"><a href="#cb55-344" aria-hidden="true" tabindex="-1"></a>ctrl.vs.acute.sig<span class="sc">$</span>status <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.negative</span>(ctrl.vs.acute.sig<span class="sc">$</span>log2FoldChange),<span class="st">"down"</span>,<span class="st">"up"</span>)</span>
<span id="cb55-345"><a href="#cb55-345" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(ctrl.vs.acute.sig<span class="sc">$</span>status)</span>
<span id="cb55-346"><a href="#cb55-346" aria-hidden="true" tabindex="-1"></a>ctrl.vs.early.sig<span class="sc">$</span>status <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.negative</span>(ctrl.vs.early.sig<span class="sc">$</span>log2FoldChange),<span class="st">"down"</span>,<span class="st">"up"</span>)</span>
<span id="cb55-347"><a href="#cb55-347" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(ctrl.vs.early.sig<span class="sc">$</span>status)</span>
<span id="cb55-348"><a href="#cb55-348" aria-hidden="true" tabindex="-1"></a>ctrl.vs.late.sig<span class="sc">$</span>status <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.negative</span>(ctrl.vs.late.sig<span class="sc">$</span>log2FoldChange),<span class="st">"down"</span>,<span class="st">"up"</span>)</span>
<span id="cb55-349"><a href="#cb55-349" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(ctrl.vs.late.sig<span class="sc">$</span>status)</span>
<span id="cb55-350"><a href="#cb55-350" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb55-351"><a href="#cb55-351" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-352"><a href="#cb55-352" aria-hidden="true" tabindex="-1"></a>The patients with Acute COVID showed highest number of DEGs (<span class="in">`571 down-regulated and 350 up-regulated`</span>) followed by Early Recovery patients (<span class="in">`317 down-regulated 378 up-regulated`</span>). This was oppose to what paper observed i.e. more DEGs in Early recovery and less in Acute Infection patients. The Late recovery patients showed very few DEGs (<span class="in">` 26 down 11 up`</span>) which is in corcordance with the paper.</span>
<span id="cb55-353"><a href="#cb55-353" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-354"><a href="#cb55-354" aria-hidden="true" tabindex="-1"></a>As authors reported, we also found down-regulation of genes related to key immune pathways, such as antigenic presentation, innate immune responses and MAPK and NF-κB signalling (JUNB, ATF3, NFkB2) in monocytes from patients suffering from Acute COVID-19. In addition, we also didn't observe the upregulation of immunosuppressive genes such as ARG1, IL-10, or IDO1 in Acute Patients which is in concordance with what the paper has reported.</span>
<span id="cb55-355"><a href="#cb55-355" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-356"><a href="#cb55-356" aria-hidden="true" tabindex="-1"></a>Like authors mentioned, for Ctrl vs Early recovery samples we also observed upregulation of genes encoding chemokines, along with important intracellular immunomodulatory proteins and transcription factors (PPARG, FOSL1, MAFB, MAFF, FOXO3) except ATF4.</span>
<span id="cb55-357"><a href="#cb55-357" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-360"><a href="#cb55-360" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb55-361"><a href="#cb55-361" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(clusterProfiler)</span>
<span id="cb55-362"><a href="#cb55-362" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(org.Hs.eg.db)</span>
<span id="cb55-363"><a href="#cb55-363" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GeneStructureTools)</span>
<span id="cb55-364"><a href="#cb55-364" aria-hidden="true" tabindex="-1"></a>a <span class="ot">&lt;-</span> <span class="fu">enrichGO</span>(<span class="at">gene =</span> <span class="fu">unique</span>(<span class="fu">removeVersion</span>(ctrl.vs.acute.sig[ctrl.vs.acute.sig<span class="sc">$</span>status<span class="sc">==</span><span class="st">"up"</span>,]<span class="sc">$</span>Geneid)),</span>
<span id="cb55-365"><a href="#cb55-365" aria-hidden="true" tabindex="-1"></a><span class="at">OrgDb =</span> org.Hs.eg.db, <span class="at">keyType =</span> <span class="st">"ENSEMBL"</span>,</span>
<span id="cb55-366"><a href="#cb55-366" aria-hidden="true" tabindex="-1"></a><span class="at">ont =</span> <span class="st">"BP"</span>,</span>
<span id="cb55-367"><a href="#cb55-367" aria-hidden="true" tabindex="-1"></a><span class="at">pAdjustMethod =</span> <span class="st">"BH"</span>,</span>
<span id="cb55-368"><a href="#cb55-368" aria-hidden="true" tabindex="-1"></a><span class="at">pvalueCutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb55-369"><a href="#cb55-369" aria-hidden="true" tabindex="-1"></a><span class="at">qvalueCutoff =</span> <span class="fl">0.1</span>,</span>
<span id="cb55-370"><a href="#cb55-370" aria-hidden="true" tabindex="-1"></a><span class="at">readable =</span> <span class="cn">TRUE</span>)</span>
<span id="cb55-371"><a href="#cb55-371" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-372"><a href="#cb55-372" aria-hidden="true" tabindex="-1"></a>b <span class="ot">&lt;-</span> <span class="fu">enrichGO</span>(<span class="at">gene =</span> <span class="fu">unique</span>(<span class="fu">removeVersion</span>(ctrl.vs.acute.sig[ctrl.vs.acute.sig<span class="sc">$</span>status<span class="sc">==</span><span class="st">"down"</span>,]<span class="sc">$</span>Geneid)),</span>
<span id="cb55-373"><a href="#cb55-373" aria-hidden="true" tabindex="-1"></a><span class="at">OrgDb =</span> org.Hs.eg.db, <span class="at">keyType =</span> <span class="st">"ENSEMBL"</span>,</span>
<span id="cb55-374"><a href="#cb55-374" aria-hidden="true" tabindex="-1"></a><span class="at">ont =</span> <span class="st">"BP"</span>,</span>
<span id="cb55-375"><a href="#cb55-375" aria-hidden="true" tabindex="-1"></a><span class="at">pAdjustMethod =</span> <span class="st">"BH"</span>,</span>
<span id="cb55-376"><a href="#cb55-376" aria-hidden="true" tabindex="-1"></a><span class="at">pvalueCutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb55-377"><a href="#cb55-377" aria-hidden="true" tabindex="-1"></a><span class="at">qvalueCutoff =</span> <span class="fl">0.1</span>,</span>
<span id="cb55-378"><a href="#cb55-378" aria-hidden="true" tabindex="-1"></a><span class="at">readable =</span> <span class="cn">TRUE</span>)</span>
<span id="cb55-379"><a href="#cb55-379" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb55-380"><a href="#cb55-380" aria-hidden="true" tabindex="-1"></a><span class="co">#dotplot</span></span>
<span id="cb55-381"><a href="#cb55-381" aria-hidden="true" tabindex="-1"></a>plt1 <span class="ot">&lt;-</span> <span class="fu">dotplot</span>(a, <span class="at">showCategory =</span> <span class="dv">10</span>,</span>
<span id="cb55-382"><a href="#cb55-382" aria-hidden="true" tabindex="-1"></a><span class="at">font.size =</span> <span class="dv">10</span>,</span>
<span id="cb55-383"><a href="#cb55-383" aria-hidden="true" tabindex="-1"></a><span class="at">title =</span> <span class="st">"Acute.vs.Healthy (Upreg)"</span>)</span>
<span id="cb55-384"><a href="#cb55-384" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-385"><a href="#cb55-385" aria-hidden="true" tabindex="-1"></a>plt2 <span class="ot">&lt;-</span> <span class="fu">dotplot</span>(b, <span class="at">showCategory =</span> <span class="dv">10</span>,</span>
<span id="cb55-386"><a href="#cb55-386" aria-hidden="true" tabindex="-1"></a><span class="at">font.size =</span> <span class="dv">10</span>,</span>
<span id="cb55-387"><a href="#cb55-387" aria-hidden="true" tabindex="-1"></a><span class="at">title =</span> <span class="st">"Acute.vs.Healthy (Downreg)"</span>)</span>
<span id="cb55-388"><a href="#cb55-388" aria-hidden="true" tabindex="-1"></a>ggpubr<span class="sc">::</span><span class="fu">ggarrange</span>(plt1,plt2,<span class="at">labels =</span> <span class="st">"AUTO"</span>)</span>
<span id="cb55-389"><a href="#cb55-389" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-390"><a href="#cb55-390" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb55-391"><a href="#cb55-391" aria-hidden="true" tabindex="-1"></a>ORA gave many over-represented biological terms for Acute DEGs but nothing significant for Eary Recovery Samples.</span>
<span id="cb55-394"><a href="#cb55-394" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb55-395"><a href="#cb55-395" aria-hidden="true" tabindex="-1"></a>a <span class="ot">&lt;-</span> <span class="fu">enrichGO</span>(<span class="at">gene =</span> <span class="fu">unique</span>(<span class="fu">removeVersion</span>(ctrl.vs.early.sig[ctrl.vs.early.sig<span class="sc">$</span>status<span class="sc">==</span><span class="st">"up"</span>,]<span class="sc">$</span>Geneid)),</span>
<span id="cb55-396"><a href="#cb55-396" aria-hidden="true" tabindex="-1"></a><span class="at">OrgDb =</span> org.Hs.eg.db, <span class="at">keyType =</span> <span class="st">"ENSEMBL"</span>,</span>
<span id="cb55-397"><a href="#cb55-397" aria-hidden="true" tabindex="-1"></a><span class="at">ont =</span> <span class="st">"BP"</span>,</span>
<span id="cb55-398"><a href="#cb55-398" aria-hidden="true" tabindex="-1"></a><span class="at">pAdjustMethod =</span> <span class="st">"BH"</span>,</span>
<span id="cb55-399"><a href="#cb55-399" aria-hidden="true" tabindex="-1"></a><span class="at">pvalueCutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb55-400"><a href="#cb55-400" aria-hidden="true" tabindex="-1"></a><span class="at">qvalueCutoff =</span> <span class="fl">0.1</span>,</span>
<span id="cb55-401"><a href="#cb55-401" aria-hidden="true" tabindex="-1"></a><span class="at">readable =</span> <span class="cn">TRUE</span>)</span>
<span id="cb55-402"><a href="#cb55-402" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-403"><a href="#cb55-403" aria-hidden="true" tabindex="-1"></a>b <span class="ot">&lt;-</span> <span class="fu">enrichGO</span>(<span class="at">gene =</span> <span class="fu">unique</span>(<span class="fu">removeVersion</span>(ctrl.vs.early.sig[ctrl.vs.early.sig<span class="sc">$</span>status<span class="sc">==</span><span class="st">"down"</span>,]<span class="sc">$</span>Geneid)),</span>
<span id="cb55-404"><a href="#cb55-404" aria-hidden="true" tabindex="-1"></a><span class="at">OrgDb =</span> org.Hs.eg.db, <span class="at">keyType =</span> <span class="st">"ENSEMBL"</span>,</span>
<span id="cb55-405"><a href="#cb55-405" aria-hidden="true" tabindex="-1"></a><span class="at">ont =</span> <span class="st">"BP"</span>,</span>
<span id="cb55-406"><a href="#cb55-406" aria-hidden="true" tabindex="-1"></a><span class="at">pAdjustMethod =</span> <span class="st">"BH"</span>,</span>
<span id="cb55-407"><a href="#cb55-407" aria-hidden="true" tabindex="-1"></a><span class="at">pvalueCutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb55-408"><a href="#cb55-408" aria-hidden="true" tabindex="-1"></a><span class="at">qvalueCutoff =</span> <span class="fl">0.1</span>,</span>
<span id="cb55-409"><a href="#cb55-409" aria-hidden="true" tabindex="-1"></a><span class="at">readable =</span> <span class="cn">TRUE</span>)</span>
<span id="cb55-410"><a href="#cb55-410" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb55-411"><a href="#cb55-411" aria-hidden="true" tabindex="-1"></a><span class="co">#dotplot</span></span>
<span id="cb55-412"><a href="#cb55-412" aria-hidden="true" tabindex="-1"></a><span class="co"># plt1 &lt;- dotplot(a, showCategory = 10,</span></span>
<span id="cb55-413"><a href="#cb55-413" aria-hidden="true" tabindex="-1"></a><span class="co"># font.size = 10,</span></span>
<span id="cb55-414"><a href="#cb55-414" aria-hidden="true" tabindex="-1"></a><span class="co"># title = "Early Recov.vs.Healthy (Upreg)")</span></span>
<span id="cb55-415"><a href="#cb55-415" aria-hidden="true" tabindex="-1"></a><span class="do">## No-enrichment was observed for upregulated genes</span></span>
<span id="cb55-416"><a href="#cb55-416" aria-hidden="true" tabindex="-1"></a>plt2 <span class="ot">&lt;-</span> <span class="fu">dotplot</span>(b, <span class="at">showCategory =</span> <span class="dv">10</span>,</span>
<span id="cb55-417"><a href="#cb55-417" aria-hidden="true" tabindex="-1"></a><span class="at">font.size =</span> <span class="dv">10</span>,</span>
<span id="cb55-418"><a href="#cb55-418" aria-hidden="true" tabindex="-1"></a><span class="at">title =</span> <span class="st">"Early Recov.vs.Healthy (Downreg)"</span>)</span>
<span id="cb55-419"><a href="#cb55-419" aria-hidden="true" tabindex="-1"></a>ggpubr<span class="sc">::</span><span class="fu">ggarrange</span>(plt2,<span class="at">labels =</span> <span class="st">"AUTO"</span>)</span>
<span id="cb55-420"><a href="#cb55-420" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb55-421"><a href="#cb55-421" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-422"><a href="#cb55-422" aria-hidden="true" tabindex="-1"></a><span class="fu">### GSEA</span></span>
<span id="cb55-423"><a href="#cb55-423" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-424"><a href="#cb55-424" aria-hidden="true" tabindex="-1"></a>On performing GSEA analysis however, we got similar pathways to be enriched as shown by authors in the paper such as mitochondrial gene expression being enriched profoundly Acute vs. Healthy (AvH) (padj=8.684443e-06) relative to Early recovery cohort (EvH) (padj=0.013; not shown in figure). Instead of term "viral gene expression" we have related term "viral life cycle" in EvH (padj=5.931676e-05) but have smaller p-value in  AvH ("response to virus" (padj=0.04712597)).</span>
<span id="cb55-427"><a href="#cb55-427" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb55-428"><a href="#cb55-428" aria-hidden="true" tabindex="-1"></a>rank.wt <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(ctrl.vs.acute<span class="sc">$</span>pvalue) <span class="sc">*</span> <span class="fu">sign</span>(ctrl.vs.acute<span class="sc">$</span>log2FoldChange)</span>
<span id="cb55-429"><a href="#cb55-429" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(rank.wt) <span class="ot">&lt;-</span> <span class="fu">removeVersion</span>(<span class="fu">rownames</span>(ctrl.vs.acute))</span>
<span id="cb55-430"><a href="#cb55-430" aria-hidden="true" tabindex="-1"></a>rank.wt <span class="ot">&lt;-</span> <span class="fu">sort</span>(rank.wt,<span class="at">decreasing =</span> T)</span>
<span id="cb55-431"><a href="#cb55-431" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-432"><a href="#cb55-432" aria-hidden="true" tabindex="-1"></a>a <span class="ot">&lt;-</span> <span class="fu">gseGO</span>(<span class="at">geneList=</span>,rank.wt,</span>
<span id="cb55-433"><a href="#cb55-433" aria-hidden="true" tabindex="-1"></a><span class="at">ont =</span><span class="st">"BP"</span>,</span>
<span id="cb55-434"><a href="#cb55-434" aria-hidden="true" tabindex="-1"></a><span class="at">keyType =</span> <span class="st">"ENSEMBL"</span>,</span>
<span id="cb55-435"><a href="#cb55-435" aria-hidden="true" tabindex="-1"></a><span class="at">minGSSize =</span> <span class="dv">3</span>,</span>
<span id="cb55-436"><a href="#cb55-436" aria-hidden="true" tabindex="-1"></a><span class="at">maxGSSize =</span> <span class="dv">800</span>,</span>
<span id="cb55-437"><a href="#cb55-437" aria-hidden="true" tabindex="-1"></a><span class="at">pvalueCutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb55-438"><a href="#cb55-438" aria-hidden="true" tabindex="-1"></a><span class="at">verbose =</span> <span class="cn">TRUE</span>,</span>
<span id="cb55-439"><a href="#cb55-439" aria-hidden="true" tabindex="-1"></a><span class="at">OrgDb =</span> org.Hs.eg.db,</span>
<span id="cb55-440"><a href="#cb55-440" aria-hidden="true" tabindex="-1"></a><span class="at">pAdjustMethod =</span> <span class="st">"BH"</span>)</span>
<span id="cb55-441"><a href="#cb55-441" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-442"><a href="#cb55-442" aria-hidden="true" tabindex="-1"></a>plt1 <span class="ot">&lt;-</span> <span class="fu">dotplot</span>(a, <span class="at">showCategory =</span> <span class="dv">10</span>,</span>
<span id="cb55-443"><a href="#cb55-443" aria-hidden="true" tabindex="-1"></a>        <span class="at">font.size =</span> <span class="dv">10</span>,</span>
<span id="cb55-444"><a href="#cb55-444" aria-hidden="true" tabindex="-1"></a>        <span class="at">title =</span> <span class="st">"Acute.vs.Healthy"</span>)</span>
<span id="cb55-445"><a href="#cb55-445" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-446"><a href="#cb55-446" aria-hidden="true" tabindex="-1"></a>rank.wt <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(ctrl.vs.early<span class="sc">$</span>pvalue) <span class="sc">*</span> <span class="fu">sign</span>(ctrl.vs.early<span class="sc">$</span>log2FoldChange)</span>
<span id="cb55-447"><a href="#cb55-447" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(rank.wt) <span class="ot">&lt;-</span> <span class="fu">removeVersion</span>(<span class="fu">rownames</span>(ctrl.vs.early))</span>
<span id="cb55-448"><a href="#cb55-448" aria-hidden="true" tabindex="-1"></a>rank.wt <span class="ot">&lt;-</span> <span class="fu">sort</span>(rank.wt,<span class="at">decreasing =</span> T)</span>
<span id="cb55-449"><a href="#cb55-449" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-450"><a href="#cb55-450" aria-hidden="true" tabindex="-1"></a>b <span class="ot">&lt;-</span> <span class="fu">gseGO</span>(<span class="at">geneList=</span>,rank.wt,</span>
<span id="cb55-451"><a href="#cb55-451" aria-hidden="true" tabindex="-1"></a><span class="at">ont =</span><span class="st">"BP"</span>,</span>
<span id="cb55-452"><a href="#cb55-452" aria-hidden="true" tabindex="-1"></a><span class="at">keyType =</span> <span class="st">"ENSEMBL"</span>,</span>
<span id="cb55-453"><a href="#cb55-453" aria-hidden="true" tabindex="-1"></a><span class="at">minGSSize =</span> <span class="dv">3</span>,</span>
<span id="cb55-454"><a href="#cb55-454" aria-hidden="true" tabindex="-1"></a><span class="at">maxGSSize =</span> <span class="dv">800</span>,</span>
<span id="cb55-455"><a href="#cb55-455" aria-hidden="true" tabindex="-1"></a><span class="at">pvalueCutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb55-456"><a href="#cb55-456" aria-hidden="true" tabindex="-1"></a><span class="at">verbose =</span> <span class="cn">TRUE</span>,</span>
<span id="cb55-457"><a href="#cb55-457" aria-hidden="true" tabindex="-1"></a><span class="at">OrgDb =</span> org.Hs.eg.db,</span>
<span id="cb55-458"><a href="#cb55-458" aria-hidden="true" tabindex="-1"></a><span class="at">pAdjustMethod =</span> <span class="st">"BH"</span>)</span>
<span id="cb55-459"><a href="#cb55-459" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-460"><a href="#cb55-460" aria-hidden="true" tabindex="-1"></a>plt2 <span class="ot">&lt;-</span> <span class="fu">dotplot</span>(b, <span class="at">showCategory =</span> <span class="dv">10</span>,</span>
<span id="cb55-461"><a href="#cb55-461" aria-hidden="true" tabindex="-1"></a>        <span class="at">font.size =</span> <span class="dv">10</span>,</span>
<span id="cb55-462"><a href="#cb55-462" aria-hidden="true" tabindex="-1"></a>        <span class="at">title =</span> <span class="st">"Early Recov.vs.Healthy"</span>)</span>
<span id="cb55-463"><a href="#cb55-463" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-464"><a href="#cb55-464" aria-hidden="true" tabindex="-1"></a>ggpubr<span class="sc">::</span><span class="fu">ggarrange</span>(plt1,plt2,<span class="at">labels =</span> <span class="st">"AUTO"</span>)</span>
<span id="cb55-465"><a href="#cb55-465" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb55-466"><a href="#cb55-466" aria-hidden="true" tabindex="-1"></a>Results from paper are shown below:</span>
<span id="cb55-467"><a href="#cb55-467" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-468"><a href="#cb55-468" aria-hidden="true" tabindex="-1"></a><span class="al">![](jci.insight.154183.f7.jpg)</span></span>
<span id="cb55-469"><a href="#cb55-469" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-470"><a href="#cb55-470" aria-hidden="true" tabindex="-1"></a><span class="fu">## Conclusion</span></span>
<span id="cb55-471"><a href="#cb55-471" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-472"><a href="#cb55-472" aria-hidden="true" tabindex="-1"></a>In summary, we were able to reproduce several major outcomes of the JCI paper but there was discord in number of DEGs when Acute and Early Recovery RNASeq data was compared to control. This could be due to number of reasons enlisted in \@ref(tab:label). This should not be surprising because there are several ways to skin a cat and the outcome will differ from time to time based on which instruments were used.</span>
<span id="cb55-473"><a href="#cb55-473" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-474"><a href="#cb55-474" aria-hidden="true" tabindex="-1"></a><span class="fu"># Session {-}</span></span>
<span id="cb55-475"><a href="#cb55-475" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-478"><a href="#cb55-478" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb55-479"><a href="#cb55-479" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span>
<span id="cb55-480"><a href="#cb55-480" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->




</body></html>