---
title: "Reproducing Publically Available Monocyte Study"
author:
  - name: Rohit Satyam
    orcid: 0000-0002-9311-4992
    email: rohit.satyam@kaust.edu.sa
    affiliations:
      - name: King Abdullah University of Science & Technology, SA
        address: 4700, KAUST
        city: Thuwal
        state: Jeedah
        postal-code: 23955
        
clean: false
date: "`r format(Sys.Date(), '%B %d, %Y')`"

vignette: >
 %\VignetteIndexEntry{Reproducing Publically Available Monocyte Study}
 %\VignetteEngine{knitr::rmarkdown}
 %\VignetteEncoding{UTF-8}  
 %\VignetteDepends{BiocStyle}
 %\VignetteAuthor{Rohit Satyam}

output: 
    BiocStyle::html_document:
      toc: true
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning = F, message = F, cache = T,cache.lazy = FALSE)
options(knitr.duplicate.label = "allow")
```
## Background

When there is a pathogenic infection in patients, WARS1 (Tryptophanyl tRNA synthetase 1) is promptly released from monocytes into the extracellular space and triggers series of events. These includes turning on TLR2 and TLR4/myeloid differentiation factor-2 (MD-2) signaling to boost innate inflammatory responses, the production of cytokines and chemokines such as TNF-α, IL-6, IL-8, and MIP-1α and neutrophil infiltration. 

Patients with sepsis have been found to have high levels of WARS1 in their blood and therefore it has also been reported to be a biomarker for predicting death in sepsis. Authors wanted to see if WARS1 can also be used as to identify sever sepsis cases where patient experience cytokine storm as well.

They found that WARS1 levels can be used to stratify patients with sepsis and septic animals associated with hyperinflammation and early mortality. WARS1 substantially upregulates a cluster of genes associated with hyperinflammatory sepsis, and rWARS1 injection accelerates hypercytokinemic septic death in mice.
## Bulk RNASeq Data analysis

We procured the bulk samples from [GSE243125](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE243125) (2 normal and 3 WARS1-stimulated PBMCs) and was uniformly processed using my in-house RNASeq analysis pipeline called [RNAGrinder](https://github.com/Rohit-Satyam/RNAgrinder) (v1.0.1). We ran MultiQC on STAR output to get the sense of data quality post alignment. We found that the data was stranded and that reads came from Reverse Strand. We will therefore take 4th column of the STAR Gene count matrices.

```{r}
library(dplyr)
library(DESeq2)
library(NOISeq)
```

```{r}
setwd('/data/coursework2024/week1/study_monocyte/results/results')
file_paths <- list.files(path = ".",pattern = "_ReadsPerGene.out.tab")

# Initialize an empty list to store the data
files<-t(plyr::ldply(lapply(file_paths, function(x){
  read.table(x, skip=4)[,4]
})))

colnames(files) <- gsub("_ReadsPerGene.out.tab","",file_paths)
rownames(files) <- read.table(file_paths[1], skip=4)[,1]

## reading metadata
meta <- read.csv("/data/coursework2024/week1/study_monocyte/meta.tsv", sep = "\t", header = T)
meta <- meta[, c(2,5)]
meta$disease_state <- c(rep("WARS1_Stim",3),rep("Saline_Normal",2))

```

For making NOIseq ExpressionSet object, one can use the code given below to get the coordinates and GC% but since different database have different chromosome indexing (0-based or 1-based) this might not be ideal.

```{r, eval=FALSE}
## Using BioMart
library(biomaRt)
ensembl <- useEnsembl(biomart = "genes", dataset = "hsapiens_gene_ensembl")

#View(listAttributes(ensembl))
## Get the coordinates of genes to get length and GC content 
attributes <- c("ensembl_gene_id", "chromosome_name", "start_position","end_position","percentage_gene_gc_content")
gene_info <- getBM(attributes = attributes, 
                   filters = "ensembl_gene_id", 
                   values = GeneStructureTools::removeVersion(rownames(files)[1:4]), 
                   mart = ensembl)
```

Let's try to procure the same information using our own GTF and FASTA file that were used for mapping the reads. This will enable us to use this code for any other non-model organism for which such information can not be obtained from BioMart.

```{r, eval=TRUE}
library(GenomicRanges)
library(rtracklayer)
library(Rsamtools)

# Load GTF file
GTFfile <- "/data/coursework2024/week1/resource/gencode.v45.primary_assembly.annotation.gtf"
GTF <- import.gff(GTFfile, format="gtf", genome="GRCh38", feature.type="gene")

# Reduce to merge overlapping features and simplify data structure
gene_ids <- mcols(GTF)$gene_id
gene_biotypes <- mcols(GTF)$gene_type

## reading the FASTA reference
FASTAfile = "/data/coursework2024/week1/resource/GRCh38.primary_assembly.genome.fa"
FASTA <- FaFile(FASTAfile)
seqs <- getSeq(FASTA, GTF)  # Use original GTF ranges to ensure correct sequence retrieval
gc_counts <- letterFrequency(seqs, letters = c("G", "C"), as.prob = FALSE)
lengths <- width(GTF)
gc_content <- rowSums(gc_counts) / lengths

# Extract necessary information (gene ID and gene biotype)
gene_info <- data.frame(
  gene_id = gene_ids,
  gene_biotype = gene_biotypes,
  gc=gc_content,name=GTF$gene_name,
  length=lengths,starts=start(GTF),ends=end(GTF),
  chr=seqnames(GTF),stringsAsFactors = FALSE)
saveRDS(gene_info, "gene_info.rds")
```

Now, once we have all the attributes of all the genes, let's compare the if the order of the genes in `gene_info` is same as the counts files we read.

```{r, eval=T}
table(rownames(files)==gene_info$gene_id)
colnames(files) == meta$run_accession
rownames(meta)<-meta$run_accession
meta <- meta[colnames(files),]
rownames(gene_info)<-gene_info$gene_id
## making expressionSet object to run noiseq functions
data_NOISEQ <- readData(data = files,
                        length=setNames(gene_info$length, gene_info$gene_id),
                        gc=setNames(gene_info$gc, gene_info$gene_id),
                        biotype=setNames(gene_info$gene_biotype, gene_info$gene_id),
                        chromosome = gene_info[,c("chr","starts","ends")],
                        factors = meta)

save(data_NOISEQ,files,gene_info,file="GSE243125_step1.Rda")
```

```{r}
## Loading all the 
#load("GSE243125_step1.Rda")
myexplodata <- dat(data_NOISEQ, type = "biodetection")
explo.plot(myexplodata, plottype = "persample")

par(mfrow = c(1, 2))
explo.plot(myexplodata, samples = c(1, 2), toplot = "protein_coding", plottype = "comparison")


mycountsbio = dat(data_NOISEQ, factor = NULL, type = "countsbio")
explo.plot(mycountsbio, toplot = 1, samples = 1, plottype = "boxplot")

mysaturation = dat(data_NOISEQ, k = 0, ndepth = 7, type = "saturation")
explo.plot(mysaturation, toplot = 1, samples = 1:2, yleftlim = NULL, yrightlim = NULL)
explo.plot(mysaturation, toplot = "protein_coding", samples = 1:4)

explo.plot(mycountsbio, toplot = "protein_coding", samples = NULL, plottype = "boxplot")

explo.plot(mycountsbio, toplot = 1, samples = NULL, plottype = "barplot")

mylengthbias = dat(data_NOISEQ, factor = "disease_state", type = "lengthbias")
explo.plot(mylengthbias, samples = NULL, toplot = "global")

myGCbias = dat(data_NOISEQ, factor = "disease_state", type = "GCbias")
explo.plot(myGCbias, samples = NULL, toplot = "global")

#mycd = dat(data_NOISEQ, type = "cd", norm = FALSE, refColumn = 1)
#explo.plot(mycd,samples = 1:12)

myPCA = dat(data_NOISEQ, type = "PCA")
## To match paper's label
#myPCA@dat$factors$disease_state <- mgsub::mgsub(myPCA@dat$factors$disease_state,unique(myPCA@dat$factors$disease_state),c("Late Recovery","Early Recovery","Acute","Control"))
explo.plot(myPCA, factor = "disease_state")

## Since we have more than two condition, the linew below will throw an error
#QCreport(data_NOISEQ, samples = NULL, factor = "disease_state", norm = FALSE)
```

### Normalization and differential expression with DESeq

NOISeq offers different flavors of normalization such as RPKM, UQ and TMM. The code to do this is given below.

```{r, message=FALSE, warning=FALSE, eval=F}

## Normalisation approached provided by NOISeq includes RPKM, Upper Quartile and TMM (Trimmed Mean of M)
myRPKM = rpkm(assayData(data_NOISEQ)$exprs, long = data_NOISEQ@featureData@data$Length, k = 0, lc = 1)
myUQUA = uqua(assayData(data_NOISEQ)$exprs, long = data_NOISEQ@featureData@data$Length, lc = 0.5, k = 0)
myTMM = tmm(assayData(data_NOISEQ)$exprs, long = 1000, lc = 0)
```

But since we wanna use DESeq2 we will use the raw counts and perform gene filtering before carrying out differential analysis.

```{r}
## Make DESeqDataSet object
GSE243125_DESeq2 <- DESeqDataSetFromMatrix(countData = data_NOISEQ@assayData$exprs,
                                           colData = data_NOISEQ@phenoData@data,
                                           design = ~ disease_state)
GSE243125_DESeq2$disease_state<-relevel(GSE243125_DESeq2$disease_state,ref = "Saline_Normal")

# Filtering genes with low counts
dim(GSE243125_DESeq2)

## Adding gene names and metadata
rowData(GSE243125_DESeq2) <- data_NOISEQ@featureData@data
## Adding Gene symbols
mcols(GSE243125_DESeq2)$names <- GTF$gene_name[match(GTF$gene_id,rownames(data_NOISEQ@featureData@data) )]

## Finding the smallest group size
smallestGroupSize <- min(table(GSE243125_DESeq2$disease_state))
smallestGroupSize
# remove genes that do not have counts >=10 in atleast 6 sample
keep <- rowSums(counts(GSE243125_DESeq2) >= 10) >= smallestGroupSize
GSE243125_DESeq2_F <- GSE243125_DESeq2[keep,] # 14,055 x 34
dim(GSE243125_DESeq2_F)
GSE243125_DESeq2_F$labels <- as.character(GSE243125_DESeq2_F$disease_state)
#GSE243125_DESeq2_F$labels <- mgsub::mgsub(GSE243125_DESeq2_F$labels,unique(GSE243125_DESeq2_F$labels),c("Late Recovery","Early Recovery","Acute","Control"))
## making PCA using VST normalised data
vsd <- vst(GSE243125_DESeq2_F, blind=TRUE,fitType = "parametric")
plotPCA(vsd, intgroup=c("labels"), returnData=FALSE)+ggplot2::theme_minimal()+
    ggplot2::stat_ellipse()
## Differential expression analysis
GSE243125_DESeq2_F<- DESeq(GSE243125_DESeq2_F)

#results(dds, contrast=c("condition","treated","untreated"))
ctrl.vs.stim <- results(GSE243125_DESeq2_F, contrast=c("disease_state","WARS1_Stim","Saline_Normal"))
```
# Understanding the DE results

# plot MA

```{r}
resultsNames(GSE243125_DESeq2_F)
ctrl.vs.stim <- lfcShrink(GSE243125_DESeq2_F, coef = 2)

## table(rownames(ctrl.vs.stim)==rownames(rowData(GSE243125_DESeq2_F)))
ctrl.vs.stim$symbol <- rowData(GSE243125_DESeq2_F)$names

  
plotMA(ctrl.vs.stim, ylim=c(-2,2))

## Filter DEGs
#filter DEGs: |FC| > 2 and FDR < 0.05 i.e. LFC value of 1
gtools::logratio2foldchange(1)
ctrl.vs.stim.sig <- subset(as.data.frame(ctrl.vs.stim) %>%tibble::rownames_to_column("Geneid"), padj <= 0.05 & abs(log2FoldChange)>1)

res <- list("Ctrl vs Stimulated"=ctrl.vs.stim.sig)
writexl::write_xlsx(res,"DEGs.xlsx")
# Why to shrink: it looks at the largest fold changes that are not due 
# to low counts and uses these to inform a prior distribution. 
# So the large fold changes from genes with lots of statistical information are 
# not shrunk, while the imprecise fold changes are shrunk. This allows you to 
# compare all estimated LFC across experiments, for example, which is not really
# feasible without the use of a prior. 
# Michael Love https://support.bioconductor.org/p/77461/
```

**Interpretation:** The MA plot provides a quick visual assessment of the differential expression analysis. It helps to identify genes that are significantly differentially expressed (blue points) and to see how the fold changes distribute across genes with different expression levels. The data points' spread along the y-axis (M) often narrows as you move to the right along the x-axis (A) because the relative change in expression (fold change) is generally more stable for genes with higher expression levels, reflecting the decreased variance associated with more abundant transcripts.

There are very few DEGs in our data

### Comparison between *Yoon Tae Kim et. al.* and our analysis.

Here we tried to reproduce their RNASeq data analysis in order to gauge to what extent we can reproduce the results. Since the paper didn't provide any PCA or QC matrices we can't comment about the preprocessing part of the data carried out by them. The bioinformatic methodology section is also quite confusing. The number of samples mentioned in the article are wrong (they say 11 samples analyzed but only 5 samples have been submitted). Moreover, STAR version and Reference genome assembly and annotation is missing.

The PCA plot shows high intra-sample variability along PC2.
**Comparing the results**

The authors used EdgeR to perform the differential expression analysis and use LFC vcalue threshold of |≥5|. But we use DESeq2 We kept the genes that passed the thresholds mentioned in the article.If we follow their specified thresholds, we get only 57 DEGs as opposed to 674 DEGs reported by authors and WARS1 is not among these 57 DEGs. Infact the WARS1 is not differentially regulated at all.

We therefore resorted to `padj <= 0.05 & abs(log2FoldChange)>1` which gives us 577 DEGs (159 up and 418 downregulated). We observed upregulated of chemokine (C-X-C motif) ligand (CXCL) genes as described by paper and genes from chemokine (C-C motif) ligand (CCL) families 

```{r}
temp <- subset(as.data.frame(ctrl.vs.stim) %>%tibble::rownames_to_column("Geneid"), padj <= 0.01 & abs(log2FoldChange)>=5)
```

```{r}
library(schoolmath)
ctrl.vs.stim.sig$status <- ifelse(is.negative(ctrl.vs.stim.sig$log2FoldChange),"down","up")
table(ctrl.vs.stim.sig$status)
```
We perform ORA and GSEA on the significant DEGs. The ORA results matches with the author's ORA results with chemotaxis terms prevalent

```{r}
library(clusterProfiler)
library(org.Hs.eg.db)
library(GeneStructureTools)
a <- enrichGO(gene = unique(removeVersion(ctrl.vs.stim.sig[ctrl.vs.stim.sig$status=="up",]$Geneid)),
OrgDb = org.Hs.eg.db, keyType = "ENSEMBL",
ont = "BP",
pAdjustMethod = "BH",
pvalueCutoff = 0.05,
qvalueCutoff = 0.1,
readable = TRUE)

b <- enrichGO(gene = unique(removeVersion(ctrl.vs.stim.sig[ctrl.vs.stim.sig$status=="down",]$Geneid)),
OrgDb = org.Hs.eg.db, keyType = "ENSEMBL",
ont = "BP",
pAdjustMethod = "BH",
pvalueCutoff = 0.05,
qvalueCutoff = 0.1,
readable = TRUE)
 
#dotplot
plt1 <- dotplot(a, showCategory = 10,
font.size = 10,
title = "WARS1_Stim.vs.Saline_Normal (Upreg)")

plt2 <- dotplot(b, showCategory = 10,
font.size = 10,
title = "WARS1_Stim.vs.Saline_Normal (Downreg)")
#ggpubr::ggarrange(plt1,plt2,labels = "AUTO")

```
![](/data/coursework2024/week1/study_monocyte/results/results/Rplot02.png)
ORA gave many over-represented biological terms such as leukocyte migration ,mononuclear cell proliferation.

**Celular component**
Like authors observed our DEGs show enrichment of terms that suggests increased gene expression in the extracellular space and plasma membrane and decreased expression of intracellular genes.
```{r}

library(clusterProfiler)
library(org.Hs.eg.db)
library(GeneStructureTools)
a <- enrichGO(gene = unique(removeVersion(ctrl.vs.stim.sig[ctrl.vs.stim.sig$status=="up",]$Geneid)),
OrgDb = org.Hs.eg.db, keyType = "ENSEMBL",
ont = "CC",
pAdjustMethod = "BH",
pvalueCutoff = 0.05,
qvalueCutoff = 0.1,
readable = TRUE)

b <- enrichGO(gene = unique(removeVersion(ctrl.vs.stim.sig[ctrl.vs.stim.sig$status=="down",]$Geneid)),
OrgDb = org.Hs.eg.db, keyType = "ENSEMBL",
ont = "CC",
pAdjustMethod = "BH",
pvalueCutoff = 0.05,
qvalueCutoff = 0.1,
readable = TRUE)
 
#dotplot
plt1 <- dotplot(a, showCategory = 10,
font.size = 10,
title = "WARS1_Stim.vs.Saline_Normal (Upreg)")

plt2 <- dotplot(b, showCategory = 10,
font.size = 10,
title = "WARS1_Stim.vs.Saline_Normal (Downreg)")
ggpubr::ggarrange(plt1,plt2,labels = "AUTO")

```

### GSEA

On performing GSEA analysis however, we got similar pathways to be enriched as shown by authors in the paper such as mitochondrial gene expression being enriched profoundly Acute vs. Saline-Normal (AvH) (padj=8.684443e-06) relative to Early recovery cohort (EvH) (padj=0.013; not shown in figure). Instead of term "viral gene expression" we have related term "viral life cycle" in EvH (padj=5.931676e-05) but have smaller p-value in  AvH ("response to virus" (padj=0.04712597)).
```{r}
rank.wt <- -log10(ctrl.vs.stim$pvalue) * sign(ctrl.vs.stim$log2FoldChange)
names(rank.wt) <- removeVersion(rownames(ctrl.vs.stim))
rank.wt <- sort(rank.wt,decreasing = T)

a <- gseGO(geneList=,rank.wt,
ont ="BP",
keyType = "ENSEMBL",
minGSSize = 3,
maxGSSize = 800,
pvalueCutoff = 0.05,
verbose = TRUE,
OrgDb = org.Hs.eg.db,
pAdjustMethod = "BH")

plt1 <- dotplot(a, showCategory = 20,
        font.size = 10,
        title = "WARS1_Stim.vs.Saline_Normal")

ggpubr::ggarrange(plt1,labels = "AUTO")
```


## Conclusion

In summary, we were not able to fully reproduce several major outcomes of the EMBO paper as there was discord in number of DEGs. When I wrote to authors, they accepted some errors in the methodology section but didn't say much on lack of reproducibility. While the ORA and GSEA and few DEG family could be corroborated in our analysis, I am not sure why the results vary so drastically.

# Session {-}

```{r}
sessionInfo()
```